import{R as o}from"./ResourceLink-82ddf6bf.js";import{H as M}from"./HeaderBodyCard-f3b47b61.js";import{a as U,d as A,b as w}from"./VCard-4d112d66.js";import{V as I,e as k}from"./VListItem-99841f34.js";import{d as V,A as x,o as n,c as f,w as e,a as t,V as P,e as m,t as v,g as s,l as T,b as a,f as g,F as h}from"./index-fa2d734a.js";import{V as u,a as c}from"./VRow-781382bd.js";import"./_plugin-vue_export-helper-c27b6911.js";const j={key:1,class:"font-weight-bold mb-0 text-white"},C={class:"font-weight-bold"},y=V({__name:"MemberCard",props:{name:{},title:{},imgUrl:{}},setup(_){const p=_,d=x(()=>{const r=p.name.split(" "),b=r[0].charAt(0),i=r[r.length-1].charAt(0);return b+i});return(l,r)=>(n(),f(w,null,{default:e(()=>[t(U,{class:"pa-2"},{default:e(()=>[t(I,{class:"grow",subtitle:l.title},{prepend:e(()=>[t(A,{size:"36px",color:"primary"},{default:e(()=>[l.imgUrl?(n(),f(P,{key:0,alt:"Avatar",src:l.imgUrl,cover:!1},null,8,["src"])):(n(),m("span",j,v(d.value),1))]),_:1})]),default:e(()=>[t(k,null,{default:e(()=>[s("span",C,v(l.name),1)]),_:1})]),_:1},8,["subtitle"])]),_:1})]),_:1}))}}),D="/unipept/assets/elixir-logo-aac6b3f9.svg",L="/unipept/assets/ugent-logo-d4e99098.png",B=s("h1",{class:"font-weight-light"}," About Unipept ",-1),S={class:"px-2 py-4"},G=s("span",null," If you use Unipept for your research, please cite: ",-1),R={class:"pl-6"},N=s("h1",{class:"font-weight-light mt-5"}," Unipept team ",-1),F=s("h1",{class:"font-weight-light mt-5"}," Former members ",-1),H=s("h1",{class:"font-weight-light mt-5"}," Supported by ",-1),J=s("h1",{class:"font-weight-light mt-5"}," Disclaimer ",-1),W=s("p",null," We make no warranties regarding the correctness of the data, and disclaim liability for damages resulting from its use. ",-1),Q=V({__name:"AboutPage",setup(_){const p=[{name:"Bart Mesuere",title:"Project lead",imgUrl:"https://avatars.githubusercontent.com/u/481872?v=4"},{name:"Pieter Verschaffelt",title:"PostDoc",imgUrl:"https://avatars.githubusercontent.com/u/9608686?v=4"},{name:"Tibo Vande Moortele",title:"PhD student",imgUrl:"https://avatars.githubusercontent.com/u/34175340?v=4"},{name:"Peter Dawyndt",title:"P.I.",imgUrl:"https://avatars.githubusercontent.com/u/5736113?v=4"},{name:"Lennart Martens",title:"P.I.",imgUrl:""}],d=[{name:"Robbert Gurdeep Singh",title:"Developer 2017-2018",imgUrl:"https://avatars.githubusercontent.com/u/662538?v=4"},{name:"Felix Van der Jeugt",title:"PhD student",imgUrl:"https://avatars.githubusercontent.com/u/994166?v=4"},{name:"Toon Willems",title:"Master's student 2013-2014",imgUrl:"https://avatars.githubusercontent.com/u/104395?v=4"},{name:"Tom Naessens",title:"Master's student 2014-2015",imgUrl:"https://avatars.githubusercontent.com/u/915130?v=4"},{name:"Stijn Seghers",title:"Master's student 2015-2016",imgUrl:"https://avatars.githubusercontent.com/u/607786?v=4"},{name:"Kevin Velghe",title:"Master's student 2015-2016",imgUrl:""},{name:"Aranka Steyaert",title:"Master's student 2016-2017",imgUrl:""},{name:"Niels De Graef",title:"Master's student 2016-2018",imgUrl:"https://avatars.githubusercontent.com/u/1382976?v=4"},{name:"Peter Vandamme",title:"Advisor",imgUrl:""},{name:"Bart Devreese",title:"Advisor",imgUrl:""}],l=[{name:"ELIXIR",imgUrl:D},{name:"UGent",imgUrl:L}];return(r,b)=>(n(),f(T,null,{default:e(()=>[t(c,null,{default:e(()=>[t(u,{sm:"12",lg:"5"},{default:e(()=>[B,s("p",null,[a(" The Unipept web application supports biodiversity and functional analysis of large and complex metaproteome samples. Its underlying index structure is designed to quickly retrieve all occurrences of a tryptic peptide in UniProt entries. Taxon-specificity of the tryptic peptide is successively derived from these occurrences using a novel lowest common ancestor approach that is robust against taxonomic misarrangements, misidentifications and inaccuracies. This application consists of two separate functionalities: the "),t(o,{to:"/tpa",router:""},{default:e(()=>[a("Tryptic Peptide Analysis")]),_:1}),a(" and the "),t(o,{to:"/mpa",router:""},{default:e(()=>[a("Metaproteomics Analysis")]),_:1}),a(". ")]),s("p",null,[a(" The initial Unipept publication can be found at "),t(o,{to:"https://pubs.acs.org/doi/10.1021/pr300576s"},{default:e(()=>[a("Unipept: tryptic peptide-based biodiversity analysis of metaproteome samples")]),_:1}),a(", a follow up publication can be found at "),t(o,{to:"https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/10.1002/pmic.201400361"},{default:e(()=>[a("The Unipept Metaproteomics Analysis Pipeline")]),_:1}),a(". The API was published as "),t(o,{to:"https://academic.oup.com/bioinformatics/article/32/11/1746/1742840?login=false"},{default:e(()=>[a("Unipept Web Services for Metaproteomics Analysis")]),_:1}),a(". ")])]),_:1}),t(u,{sm:"12",lg:"7"},{default:e(()=>[t(M,{title:"How to cite Unipept"},{default:e(()=>[s("div",S,[G,s("ul",R,[s("li",null,[a(" Unipept Desktop: Verschaffelt et al. (2021) "),t(o,{to:"https://doi.org/10.1021/acs.jproteome.0c00855"},{default:e(()=>[a(" doi.org/10.1021/acs.jproteome.0c00855 ")]),_:1})]),s("li",null,[a(" CLI 2.0: Verschaffelt et al. (2020) "),t(o,{to:"https://doi.org/10.1093/bioinformatics/btaa553"},{default:e(()=>[a(" doi.org/10.1093/bioinformatics/btaa553 ")]),_:1})]),s("li",null,[a(" CLI: Mesuere et al. (2017) "),t(o,{to:"https://doi.org/10.1016/j.jprot.2017.05.022"},{default:e(()=>[a(" doi.org/10.1016/j.jprot.2017.05.022 ")]),_:1})]),s("li",null,[a(" Peptidome Analysis: Mesuere et al. (2016) Proteomics "),t(o,{to:"https://doi.org/10.1002/pmic.201600023"},{default:e(()=>[a(" doi.org/10.1002/pmic.201600023 ")]),_:1})]),s("li",null,[a(" API: Mesuere et al. (2016) Bioinformatics "),t(o,{to:"https://doi.org/10.1093/bioinformatics/btw039"},{default:e(()=>[a(" doi.org/10.1093/bioinformatics/btw039 ")]),_:1})]),s("li",null,[a(" General: Gurdeep Singh et al. (2019) Journal of Proteome Research "),t(o,{to:"https://doi.org/10.1021/acs.jproteome.8b00716"},{default:e(()=>[a(" doi.org/10.1021/acs.jproteome.8b00716 ")]),_:1})]),s("li",null,[a(" General: Mesuere et al. (2015) Proteomics "),t(o,{to:"https://doi.org/10.1002/pmic.201400361"},{default:e(()=>[a(" doi.org/10.1002/pmic.201400361 ")]),_:1})]),s("li",null,[a(" General: Mesuere et al. (2012) J. Proteome Res. "),t(o,{to:"https://doi.org/10.1021/pr300576s"},{default:e(()=>[a(" doi.org/10.1021/pr300576s ")]),_:1})])])])]),_:1})]),_:1})]),_:1}),N,t(c,null,{default:e(()=>[(n(),m(h,null,g(p,i=>t(u,{key:i.name,sm:"6",md:"4"},{default:e(()=>[t(y,{name:i.name,title:i.title,"img-url":i.imgUrl},null,8,["name","title","img-url"])]),_:2},1024)),64))]),_:1}),F,t(c,null,{default:e(()=>[(n(),m(h,null,g(d,i=>t(u,{key:i.name,sm:"6",md:"4"},{default:e(()=>[t(y,{name:i.name,title:i.title,"img-url":i.imgUrl},null,8,["name","title","img-url"])]),_:2},1024)),64))]),_:1}),H,t(c,null,{default:e(()=>[(n(),m(h,null,g(l,i=>t(u,{key:i.name,sm:"6",md:"4"},{default:e(()=>[t(w,null,{default:e(()=>[t(U,null,{default:e(()=>[t(P,{src:i.imgUrl,height:"100px",contain:""},null,8,["src"])]),_:2},1024)]),_:2},1024)]),_:2},1024)),64))]),_:1}),J,W]),_:1}))}});export{Q as default};
