import{H as r}from"./HeaderBodyCard-f3b47b61.js";import{R as o}from"./ResourceLink-82ddf6bf.js";import{B as s,S as a}from"./Boxed-1ca540a2.js";import{I as l}from"./InlineCode-b941903d.js";import{S as p}from"./StaticAlert-03392fb6.js";import{I as c}from"./Initialism-25eec5f3.js";import{u as _}from"./useNavigation-4732dd9e.js";import{d as g,o as d,c as b,w as t,l as y,g as i,b as e,a as n,e as w,f as v,F as x,u as k,t as m}from"./index-fa2d734a.js";import{V as I}from"./VTable-b3c64698.js";import"./VCard-4d112d66.js";import"./_plugin-vue_export-helper-c27b6911.js";import"./VAlert-57bda60a.js";const R={class:"font-weight-light"},U=i("small",null,"v2.2.1",-1),C={class:"font-weight-light"},T={class:"mt-5"},S={class:"mt-3"},L=i("thead",null,[i("tr",null,[i("th",{class:"text-left"},"Resource"),i("th",{class:"text-left"},"Description")])],-1),P=["onClick"],B={style:{"white-space":"nowrap"}},M={class:"py-3"},D=i("br",null,null,-1),F=i("br",null,null,-1),G=i("br",null,null,-1),V=i("br",null,null,-1),$=i("br",null,null,-1),W=i("br",null,null,-1),q=i("br",null,null,-1),N=i("br",null,null,-1),A=i("br",null,null,-1),E=i("br",null,null,-1),H=i("br",null,null,-1),O=i("br",null,null,-1),j=i("br",null,null,-1),z=i("br",null,null,-1),J=i("br",null,null,-1),K=i("br",null,null,-1),Q=i("br",null,null,-1),X=i("br",null,null,-1),fe=g({__name:"CLIDocsOverviewPage",setup(Y){const{navigate:f}=_(),h=[{resource:"uniprot",description:"Fetches UniProt entries based on their accession numbers.",link:"/clidocs/uniprot"},{resource:"prot2pept",description:"Splits proteins into peptides based on (trypsin) digest.",link:"/clidocs/prot2pept"},{resource:"peptfilter",description:"Filters peptides based on length and amino acid occurrence.",link:"/clidocs/peptfilter"},{resource:"unipept pept2prot",description:"Returns the set of UniProt entries containing a given tryptic peptide.",link:"/clidocs/pept2prot"},{resource:"unipept pept2taxa",description:"Returns the set of taxa extracted from the UniProt entries containing a given tryptic peptide.",link:"/clidocs/pept2taxa"},{resource:"unipept pept2lca",description:"Returns the taxonomic lowest common ancestor for a given tryptic peptide.",link:"/clidocs/pept2lca"},{resource:"unipept pept2ec",description:"Returns the functional EC-numbers associated with a given tryptic peptide.",link:"/clidocs/pept2ec"},{resource:"unipept pept2go",description:"Returns the functional GO-terms associated with a given tryptic peptide.",link:"/clidocs/pept2go"},{resource:"unipept pept2interpro",description:"Returns the functional InterPro entries associated with a given tryptic peptide.",link:"/clidocs/pept2interpro"},{resource:"unipept pept2funct",description:"Returns the functional EC-numbers, GO-terms and InterPro entries associated with a given tryptic peptide.",link:"/clidocs/pept2funct"},{resource:"unipept peptinfo",description:"Returns functional information and the lowest common ancestor for a given tryptic peptide.",link:"/clidocs/peptinfo"},{resource:"unipept taxa2lca",description:"Returns the taxonomic lowest common ancestor for a given list of taxon identifiers.",link:"/clidocs/taxa2lca"},{resource:"unipept taxa2tree",description:"Returns the taxonomic tree for a given list of taxon identifiers.",link:"/clidocs/taxa2tree"},{resource:"unipept taxonomy",description:"Returns the taxonomic information for a given taxon identifier.",link:"/clidocs/taxonomy"}];return(Z,ee)=>(d(),b(y,null,{default:t(()=>[i("h1",R,[e(" Unipept "),n(c,null,{default:t(()=>[e("CLI")]),_:1}),e(" documentation "),U]),i("h3",C,[e(" The Unipept command line interface ("),n(c,null,{default:t(()=>[e("CLI")]),_:1}),e(") offers an easy way to integrate Unipept metaproteomics functionality into your data processing pipelines and scripts. ")]),i("p",T,[e(" These tools provide a command line interface to the "),n(o,{to:"/apidocs",router:""},{default:t(()=>[e("Unipept web services")]),_:1}),e(" and a few utility commands for handling proteins using the command line. All tools support fasta and plain text input, multiple output formats (csv, xml and json) and parallel web requests for improved performance. ")]),i("p",S,[e(" The Unipept command line tools are open source under the "),n(c,null,{default:t(()=>[e("MIT")]),_:1}),e(" License and all code is available on "),n(o,{to:"https://github.com/unipept/unipept-cli"},{default:t(()=>[e("Github")]),_:1}),e(". Changes between releases are listed in the "),n(o,{to:"/news/cli",router:""},{default:t(()=>[e("changelog")]),_:1}),e(". In case you have encountered an issue using these tools, have feature requests or found a bug, don't hesitate to contact us by email ("),n(o,{to:"unipept@ugent.be",mail:""},{default:t(()=>[e("unipept@ugent.be")]),_:1}),e("), or create an "),n(o,{to:"https://github.com/unipept/unipept-cli/issues"},{default:t(()=>[e("issue")]),_:1}),e(" on Github. ")]),n(r,{id:"functionality",title:"CLI functionality",class:"mt-5","large-title":""},{default:t(()=>[n(I,null,{default:t(()=>[L,i("tbody",null,[(d(),w(x,null,v(h,u=>i("tr",{key:u.resource,class:"clickable",onClick:te=>k(f)(u.link)},[i("td",B,[i("b",null,m(u.resource),1)]),i("td",M,m(u.description),1)],8,P)),64))])]),_:1})]),_:1}),n(r,{id:"installation",title:"Installation",class:"mt-5","large-title":""},{default:t(()=>[i("p",null,[e(" To use the Unipept command line tools, Ruby needs to be installed on your system. We recommend using Ruby 2.6, but all versions since Ruby 2.3 are supported. To check if you have the correct Ruby version installed, open a terminal and run "),n(l,null,{default:t(()=>[e("ruby --version")]),_:1}),e(". ")]),n(s,null,{default:t(()=>[n(a,null,{default:t(()=>[e("$")]),_:1}),e(" ruby --version "),D,e("ruby 2.6.0p0 (2018-12-25 revision 66547) [x86_64-darwin18] ")]),_:1}),n(p,{class:"mt-5",title:"Installing Ruby"},{default:t(()=>[i("p",null,[e(" If the "),n(l,null,{default:t(()=>[e("ruby --version")]),_:1}),e(" command returns command not found, Ruby is not yet installed on your system. More information on installing Ruby can be found at "),n(o,{to:"https://www.ruby-lang.org/en/installation/"},{default:t(()=>[e("https://www.ruby-lang.org/en/installation/")]),_:1}),e(". ")])]),_:1}),i("p",null,[e(" The Unipept CLI is available as a "),n(o,{to:"https://rubygems.org/gems/unipept"},{default:t(()=>[e("gem")]),_:1}),e(". This means it can easily be installed with the "),n(l,null,{default:t(()=>[e("gem install unipept")]),_:1}),e(" command: ")]),n(s,null,{default:t(()=>[n(a,null,{default:t(()=>[e("$")]),_:1}),e(" gem install unipept "),F,e("Fetching: unipept-2.2.1.gem (100%) "),G,e("Successfully installed unipept-2.2.1 "),V,e("Parsing documentation for unipept-2.2.1 "),$,e("Installing ri documentation for unipept-2.2.1 "),W,e("Done installing documentation for unipept after 0 seconds "),q,e("1 gem installed ")]),_:1}),i("p",null,[e(" After successful installation, the unipept command should be available. To check if unipept was installed correctly, run "),n(l,null,{default:t(()=>[e("unipept --version")]),_:1}),e(". This should print the version number: ")]),n(s,null,{default:t(()=>[n(a,null,{default:t(()=>[e("$")]),_:1}),e(" unipept --version "),N,e("2.2.1 ")]),_:1}),i("p",null,[e(" More information about the installed command can be found on these pages, or by running the "),n(l,null,{default:t(()=>[e("unipept -h")]),_:1}),e(" command. ")]),n(p,{class:"mt-5",title:"Permission problems"},{default:t(()=>[i("p",null,[e(" When trying to install a gem, you might run into permission problems if you don't have write access to the default installation directory. This can be fixed by doing a user install, which installs the gem in your home directory. Simply run "),n(l,null,{default:t(()=>[e("gem install unipept --user-install")]),_:1}),e(" instead of the normal installation command. More information can be found on the "),n(o,{to:"https://guides.rubygems.org/faqs/#user-install"},{default:t(()=>[e("RubyGems website")]),_:1}),e(". ")])]),_:1}),n(p,{class:"mt-5",title:"Windows support"},{default:t(()=>[i("p",null,[e(" Unipept "),n(c,null,{default:t(()=>[e("CLI")]),_:1}),e(" should work on Unix, Mac, and Windows. However, because of one of our underlying dependencies (curl), some windows users experience issues during installation. In case of such issues, we recommend using the Unix Subsystem of Windows 10 instead. More information can be found on the "),n(o,{to:"https://docs.microsoft.com/en-us/windows/wsl/about"},{default:t(()=>[e("Microsoft website")]),_:1}),e(". ")])]),_:1})]),_:1}),n(r,{id:"updates",title:"Updates",class:"mt-5","large-title":""},{default:t(()=>[i("p",null,[e(" To update the Unipept command line tools to the latest version, simply run "),n(l,null,{default:t(()=>[e("gem update unipept")]),_:1}),e(". The changes between releases are listed in the "),n(o,{to:"/news",router:""},{default:t(()=>[e("changelog")]),_:1}),e(". ")]),n(s,null,{default:t(()=>[n(a,null,{default:t(()=>[e("$")]),_:1}),e(" gem update unipept "),A,e("Updating installed gems "),E,e("Updating unipept "),H,e("Fetching: unipept-2.2.1.gem (100%) "),O,e("Successfully installed unipept-2.2.1 "),j,e("Parsing documentation for unipept-2.2.1 "),z,e("Installing ri documentation for unipept-2.2.1 "),J,e("Installing darkfish documentation for unipept-2.2.1 "),K,e("Done installing documentation for unipept after 0 seconds "),Q,e("Gems updated: unipept ")]),_:1})]),_:1}),n(r,{id:"configuration",title:"Configuration",class:"mt-5","large-title":""},{default:t(()=>[i("p",null,[e(" The Unipept command line tools require no additional configuration and can be used immediately after installation. By default, the public Unipept server will be used for all commands. If you have a local Unipept server running, and wish to use it with the command line tools, simply run the "),n(l,null,{default:t(()=>[e("unipept config host http://local.server")]),_:1}),e(" command, where you substitute "),n(l,null,{default:t(()=>[e("http://local.server")]),_:1}),e(" with the address of your local Unipept server: ")]),n(s,null,{default:t(()=>[n(a,null,{default:t(()=>[e("$")]),_:1}),e(" unipept config host http://local.server "),X,e("host was set to http://local.server ")]),_:1}),n(p,{class:"mt-5",title:"Default settings"},{default:t(()=>[i("p",null,[e(" If you changed the host settings and wish to return to the default host, simply run "),n(l,null,{default:t(()=>[e("unipept config host http://api.unipept.ugent.be")]),_:1}),e(". ")])]),_:1})]),_:1})]),_:1}))}});export{fe as default};
