import{H as n}from"./HeaderBodyCard-3295d4c5.js";import{d as c,o as r,c as h,w as a,l as d,a as s,g as e,V as o,u as i,b as t,ar as p}from"./index-eea28a33.js";import{V as l}from"./VAlert-6315e811.js";import"./VCard-9d59c194.js";import"./_plugin-vue_export-helper-c27b6911.js";const u="/assets/desktop_overview-62a477cc.png",m="/assets/desktop_add_assay_dialog-63ea3492.png",f="/assets/desktop_configure_assays-20949fd6.png",y="/assets/desktop_taxonomic_analysis-ae7465ae.png",g="/assets/desktop_functional_analysis-64e28175.png",w="/assets/desktop_taxonomic_filter-e55994d9.png",b=e("h1",{class:"font-weight-light"}," Unipept Desktop - Single Assay Analysis ",-1),_=e("h3",{class:"font-weight-light mb-2"}," Analyse individual metaproteomics samples and browser their taxonomic and functional composition. ",-1),v={class:"d-flex justify-center"},x=e("p",null,[t(" The single assay analysis page provides you with the ability to analyse individual metaproteomics samples, browse all samples in a project and organize them within a project (into studies). This page consists of three main sections. The "),e("span",{class:"font-weight-bold"},"sidebar"),t(" on the far left, the "),e("span",{class:"font-weight-bold"},"project explorer"),t(" directly to the right of the sidebar and the "),e("span",{class:"font-weight-bold"},"content viewer"),t(". ")],-1),k=e("p",null," The sidebar allows is the same throughout all pages of the application and allows you to switch between the different functions of the application. Directly to the right of the sidebar is the project explorer which provides a view on the contents of the current project. It allows you to create studies or assays and to switch between assays. The currently selected assay will be highlighted in blue and is the one for which the taxonomic and functional composition can be viewed in the content viewer on the right. ",-1),T=e("h3",null,"Start by creating a study",-1),A=e("p",null,[t(" In order to add a new assay to the current project, you first have to make sure that at least one study exists. An assay must be associated to a study at all times. If no study exists yet, you can create one by clicking on the "),e("span",{class:"font-weight-bold"},"Add study"),t(' button in the project explorer. You will notice that a new item appears in the project explorer with the name "Study name". This is simply the default name that is given to newly create studies and can easily be changed by either double-clicking the study or by right-clicking on the study item and selecting '),e("span",{class:"font-weight-bold"},"Rename study"),t(". ")],-1),I=e("span",{class:"font-weight-bold"},"Remark:",-1),j=e("h3",{class:"mt-6"}," Adding one or more assays to a study ",-1),F=e("span",{class:"font-weight-bold"},"Add assay",-1),B=e("span",{class:"font-weight-bold"},"Empty assay",-1),D=e("span",{class:"font-weight-bold"},"Bulk import from files",-1),V={class:"my-5"},C=e("div",null,[e("span",{class:"font-weight-bold"},"Figure 1:"),t(" the assay creation dialog will guide you through the process of adding one or more assays to your project. ")],-1),U=e("span",{class:"font-weight-bold"},"Remark:",-1),O=e("h3",{class:"mt-6"}," Configuring the assays before importing ",-1),S={class:"my-5"},z=e("div",null,[e("span",{class:"font-weight-bold"},"Figure 2:"),t(" search settings for each of the new assays can also be configured in the assay creation dialog. ")],-1),P=e("p",null," It is important to note here that each assay can have its own search settings. These settings determine a set of parameters that influence the outcome of your metaproteomics analysis: ",-1),q=e("ul",null,[e("li",null,[e("span",{class:"font-weight-bold"},"Equate I/L:"),t(" Since the amino acids isoleucine and leucine share the same mass, they can be indistinguishable in a mass spectrometry experiment. By enabling this setting, Unipept will treat isoleucine and leucine as the same amino acid. ")]),e("li",null,[e("span",{class:"font-weight-bold"},"Filter duplicates:"),t(" By enabling this setting, only unique peptides will be retained. This means that if a peptide is found multiple times in the input file, only one occurrence will be retained. ")]),e("li",null,[e("span",{class:"font-weight-bold"},"Advanced missed cleavage handling:"),t(" If not all input peptides are tryptic peptides, you can enable this setting to allow for more peptides to be matched by Unipept. Enabling this setting will cause the analysis time to increase, so it is recommended only to use it when strictly necessary. ")]),e("li",null,[e("span",{class:"font-weight-bold"},"Analysis source:"),t(` The analysis source determines which protein reference database is used for the analysis of the corresponding sample. You can choose between Unipept's online services (which typically try to match the peptides against all proteins in UniProtKB) or a local database (that typically contains only a subset of the proteins in UniProtKB). Note that local databases can be constructed on the "Targeted database management" page. `)])],-1),N=e("p",{class:"mt-2"},[t(" Finally, click on "),e("span",{class:"font-weight-bold"},"Continue"),t(" to import the assays into the current project and start the metaproteomic analysis. ")],-1),E=e("p",null,' Once an analysis is finished, the main results are split into a taxonomic and a functional part. On top of the "Single assay analysis" page, you will directly see a summary of the analysis including some useful statistics. Directly underneath this summary is a card that shows all taxonomic analysis results and on the bottom is a third card that shows all functional analysis results. ',-1),Y=e("h3",{class:"mt-6"}," Taxonomic profile ",-1),R=e("p",null," In order to explore the taxonomic composition of a sample, users can browser through 4 different visualizations that each try to model the same information in a different way. This allows individual users to gain a deeper understanding of the organisms that are present in the sample. The tabs on top of the taxonomic profile card allow you to switch between the different visualizations. ",-1),H={class:"my-5"},K=e("div",null,[e("span",{class:"font-weight-bold"},"Figure 3:"),t(" the taxonomic analysis of a metaproteomics sample results in a set of interactive visualizations that can be used to explore the results. ")],-1),G=e("h3",{class:"mt-6"}," Functional profile ",-1),L=e("p",null," The functional profile of a sample can be explored in the last card on this page. A functional summary is provided for all identified Gene Ontology terms, Enzyme Commission numbers and InterPro domains. You can switch between these distinct types of functional annotations by using the tabs on top of the functional card. ",-1),J={class:"my-5"},M=e("div",null,[e("span",{class:"font-weight-bold"},"Figure 4:"),t(" the functional analysis of a metaproteomics sample results in a set of tables that allows users to explore the different functional domains. ")],-1),Q=e("h3",{class:"mt-6"}," Filtering by taxon ",-1),W=e("p",null,` One of the key features of Unipept that distinguish it from other major metaproteomics analysis tools is it's ability to keep track of the link between taxa and functions in a sample. This allows Unipept to filter all functional annotations by a specific taxon which allows you to find out which functions are performed by which set of organisms in the environment under study. In order to perform such a filtering, you can click on any of the taxa in the taxonomic visualizations. If you, for example, click the "Bacteria" node in the sunburst, the functional profile below will only take into account bacterial taxa. `,-1),X={class:"my-5"},Z=e("div",null,[e("span",{class:"font-weight-bold"},"Figure 5:"),t(" if a taxonomic filter is active, a banner on top of the page will be clearly displayed (and also allows you to reset the filter). ")],-1),le=c({__name:"DesktopSingleAssayAnalysisPage",setup($){return(ee,te)=>(r(),h(d,null,{default:a(()=>[b,_,s(n,{id:"overview",title:"Overview",class:"mb-5","large-title":""},{default:a(()=>[e("div",v,[s(o,{src:i(u),"max-width":"800",cover:!1,eager:"",class:"screenshot ma-6"},null,8,["src"])]),x,k]),_:1}),s(n,{id:"analysing_assay",title:"Analyse an assay",class:"mb-5","large-title":""},{default:a(()=>[T,A,s(l,{color:"info",variant:"outlined"},{default:a(()=>[I,t(" Please note that each study and assay must have a unique name. No two studies in one project (or two assays in a study) can have the same name. This requirement will automatically be enforced by the application. ")]),_:1}),j,e("p",null,[t(" Once a study has been created, you can add a new assay by clicking the little file icon ("),s(p,{"x-small":""},{default:a(()=>[t("mdi-file-plus-outline")]),_:1}),t(") on the right side of the study name. A dialog will pop up (see Figure 1). This dialog allows you to create one (or more!) assays at once. If you click the "),F,t(" button, you will be prompted to choose between two options. The first option ("),B,t(") allows you to manually add a single assay. You will be required to paste a list of peptides into the text area that pops up and choose a suiting name for your new assay. The second option ("),D,t(") can be used when you have a large number of peptide files that need to be imported at once. In this case, a new assay entry will be added for each input file and the name of the assays will be deducted from the provided file name. ")]),e("div",V,[s(o,{src:i(m),"max-width":"800",cover:!1,eager:"",class:"screenshot ma-6"},null,8,["src"]),C]),s(l,{color:"info",variant:"outlined",class:"mb-4"},{default:a(()=>[U,t(' Please note that the application only accepts text-based files containing one peptide per line. These files should end with the ".txt;" or ".pep" extension. If this is not the case, an error message will be displayed. ')]),_:1}),O,e("div",S,[s(o,{src:i(f),"max-width":"800",cover:!1,eager:"",class:"screenshot ma-6"},null,8,["src"]),z]),P,q,N]),_:1}),s(n,{id:"taxonomic_analysis_results",title:"Browse analysis results","large-title":""},{default:a(()=>[E,Y,R,e("div",H,[s(o,{src:i(y),"max-width":"800",cover:!1,eager:"",class:"screenshot ma-6"},null,8,["src"]),K]),G,L,e("div",J,[s(o,{src:i(g),"max-width":"800",contain:"",eager:"",class:"screenshot ma-6"},null,8,["src"]),M]),Q,W,e("div",X,[s(o,{src:i(w),"max-width":"800",contain:"",eager:"",class:"screenshot ma-6"},null,8,["src"]),Z])]),_:1})]),_:1}))}});export{le as default};
