import{R as m}from"./ResourceLink-82ddf6bf.js";import{H as u}from"./HeaderBodyCard-f3b47b61.js";import{d as A,o as V,c as S,w as s,l as h,a as t,g as e,V as i,u as a,b as n,K as f,m as C,b5 as r}from"./index-fa2d734a.js";import{a as l,V as o}from"./VRow-781382bd.js";import{a as T}from"./VMenu-46c570ce.js";import{V as O}from"./VList-6f91db5c.js";import{V as _}from"./VListItem-99841f34.js";import{V as w}from"./VAlert-57bda60a.js";import"./_plugin-vue_export-helper-c27b6911.js";import"./VCard-4d112d66.js";import"./VOverlay-a17eb942.js";import"./delay-fab8d36b.js";import"./forwardRefs-e2994a01.js";import"./VDivider-7dfa9365.js";const W="/unipept/assets/desktop_overview-62a477cc.png",g="/unipept/assets/desktop_install_windows_1-80a95ea1.png",b="/unipept/assets/desktop_install_windows_2-d6c2240a.png",y="/unipept/assets/desktop_install_windows_3-e838020b.png",k="/unipept/assets/desktop_install_windows_4-17d491d7.png",v="/unipept/assets/desktop_install_macos_1-d6cc30ef.png",I="/unipept/assets/desktop_install_macos_2-26c46cb9.png",D="/unipept/assets/desktop_install_ubuntu_1-e76ba3d2.png",x="/unipept/assets/desktop_install_ubuntu_2-cdcfcc44.png",U="/unipept/assets/desktop_install_ubuntu_3-d1f4caef.png",M="/unipept/assets/desktop_download_static_database-a0fbf715.png",L="/unipept/assets/desktop_update_notification-9b8fba01.png",P="/unipept/assets/desktop_update_changes-0d0ddd18.png",B=e("h1",{class:"font-weight-light"},[n(" Unipept Desktop documentation "),e("small",null,"v2.0.0")],-1),R=e("h3",{class:"font-weight-light mb-2"}," The Unipept Desktop application allows, in contrast to the web application, to analyse much larger samples, to store analysis results offline, extensively compare assays with each other, better manage samples and perform proteogenomics data analyses using targeted protein reference databases. ",-1),q=e("p",null," Unipept's web application provides users with the ability to analyze a metaproteomics sample and extract taxonomic and functional information from environmental samples. Our web app provides interactive visualizations to researchers and allows them to, for example, filter out all functions that are associated with a specific taxon. Due to its inherent web-based nature, the size and amount of samples that can be analysed are rather limited. ",-1),z=e("p",null," The Unipept Desktop application, however, is not limited in the amount of memory and storage that it can consume, allowing us to process datasets with much more peptides than before. This a list of all major differences between the Unipept desktop app and the web app: ",-1),H=e("ul",null,[e("li",null,[e("span",{class:"font-weight-bold"},"Support for proteogenomics:"),n(" configure and construct your own targeted protein reference databases that only contain a subset of all the proteins from the UniProtKB resource. ")]),e("li",null,[e("span",{class:"font-weight-bold"},"Organise samples in studies and projects:"),n(" it's now easier than ever to combine and organise samples into studies and projects. One project contains all information required to reconstruct an experiment and can easily be shared with other researchers. ")]),e("li",null,[e("span",{class:"font-weight-bold"},"Analyse much larger samples:"),n(" since the desktop app is not artificially limited in the amount of compute resources that it is allowed to use, it allows you to analyse samples that contain much more peptides (when compared to the web app). ")]),e("li",null,[e("span",{class:"font-weight-bold"},"Improved inter- and intra-assay comparison pipeline:"),n(" the desktop app provides an improved comparative analysis pipline that allows you to compare both features within and between samples. This can help you to improve your insight in the functional or taxonomic shift that might occur in an ecosystem over time. ")])],-1),N=e("p",null," In order to use the Unipept Desktop Application, it needs to be installed first. You can always download the most recent version for your operating system from the releases on GitHub. Direct links for the different operating systems are also available (these links will always point to the most recent version): ",-1),E=e("p",null," Most required dependencies will automatically be installed using the installation packages provided above. If a specific installer is not suitable for your current operating system (e.g. specific flavours of Linux that do not support AppImage's), you can always clone the repository and build the project yourselves. ",-1),Y=e("span",{class:"font-weight-bold"},"Warning:",-1),F=e("a",{href:"https://www.docker.com/products/docker-desktop/"},"Docker Desktop",-1),G=e("h3",null,"1. Start the installer package",-1),K=e("p",null,' Double-click the installer package to start the installation. You will be greeted by a window that asks for your explicit approval to install our application. Click "More info". ',-1),J=["href"],Q=e("h3",null,"2. Proceed with security checks",-1),X=e("p",null,` A new button appears "Run anyway". Click this button to start the installation process. Windows shows this warning as part of the "SmartScreen Security" system. This system protects your computer from running potentially unsafe apps. Windows displays this message for our application since it's very new, without a large userbase yet. The more people start to use this app, the faster it will be trusted by Microsoft. `,-1),Z=["href"],$=e("h3",null,"3. Installation process is started",-1),ee=e("p",null," The installation process is now started, and a progress bar should be visible. This process should not take longer than one minute on most modern systems. Please wait for the installer to disappear (without any errors). Once it's gone, the installation process is fully finished. ",-1),te=["href"],se=e("h3",null,"4. Installation process is done",-1),ae=e("p",null," You can now start the application, by double clicking it's shortcut on your desktop or by selecting it from the applications list in Windows. ",-1),oe=["href"],ne=e("h3",null,"1. Mount the installer image",-1),ie=e("p",null,' Double-click the downloaded "Unipept.dmg" file to mount the installer image. A new window pops up in which you need to drag the Unipept Desktop logo into your applications folder. ',-1),le=["href"],re=e("h3",null,"2. Installation process is done",-1),ce=e("p",null," Look for the installed application in Launchpad and start Unipept Desktop. ",-1),pe=["href"],de=e("h3",null,"1. Open AppImage properties",-1),ue=e("p",null,' Right-click the downloaded "Unipept.AppImage" file and select "properties". A properties window will pop up. ',-1),he=["href"],fe=e("h3",null,"2. Enable executable permissions",-1),me=e("p",null,' Click the "permissions" tab and check the box before "Allow executing file as program". Close the window. ',-1),_e=["href"],we=e("h3",null,"3. Open the application",-1),ge=e("p",null," Double-click the downloaded AppImage-file to start the application. ",-1),be=["href"],ye=e("p",null," If this is your first time using the application, we recommend to follow the steps in this guide to get you started easily and to get acquainted with the basic functionality of the application. ",-1),ke=e("h3",null,"Static database download",-1),ve=e("p",null,' A few things are automatically set up the first time you start this application. You will directly notice a dialog popping up that downloads a small SQLite-database. This database is called the "static database" and contains static information about taxa and functional annotations. Every month, a new release of this database is automatically generated. Unipept Desktop will regularly check for a new version of this database and automatically download it when necessary. ',-1),Ie={class:"d-flex justify-center"},De=e("p",null," Download and installation of this static database can take a few minutes, depending on the available speed of your internet connection. If, for some reason, the download fails, the application will automatically fallback to a (slower) online version of this database and will continue to work. A new attempt to install the database will be made when you restart the application. ",-1),xe=e("h3",{class:"mb-2"}," Opening the demo project ",-1),Ue=e("span",{class:"font-weight-bold"},"Warning:",-1),je=e("p",null,' Every assay or sample needs to be associated with a project. So, before you can start analyzing your data, you need to create a project. However, since this is the first time you use the application, we recommend you to click the "Open demo project" button. By doing so, a preconfigured project will be loaded and you can quickly browse the analysis results that are being produced by Unipept and play with some of its features. ',-1),Ae=e("h3",null,"Creating a new project",-1),Ve=e("p",null,' In order to create a project of your own, head back to the homepage of the application by clicking on the "home" button in the sidebar. Here, you will find a button to create a new project or to reopen an existing project. Click the "New project" button, select an empty folder (or create a new one) and continue. ',-1),Se=e("h3",null,"Saving a project",-1),Ce=e("p",null," Changes made to projects are automatically saved. If you close the application, the current state of the project will already be persisted and there's nothing else that needs to be done. ",-1),Te={class:"mb-6"},Oe=e("div",null,[e("span",{class:"font-weight-bold"},"Figure 1:"),n(" a notification is shown during the automatic application update process on the bottom of the main application window. ")],-1),We=e("p",null," The application will automatically be kept up-to-date. Everytime it is started, Unipept will automatically contact the update-servers and check if a new version if available. If this is the case, a notification will appear on the bottom of the application window. The notification shows the update progress and will ask you to restart the application once the update is finished. The previous version will be used as long as the app is not (manually) restarted. ",-1),Me=e("p",null,[n(" A summary of all changes that were made between the current version of the application and the previous version will always be displayed on the right side of the home page. This summary (for all versions) is also available on our "),e("a",{class:"link",href:"https://github.com/unipept/unipept-desktop/releases"},"release notes page"),n(". ")],-1),Le={class:"mb-6"},Pe=e("div",null,[e("span",{class:"font-weight-bold"},"Figure 2:"),n(" when an update is available, a small message will appear on the homepage. A dialog with a summary of all changes will pop up after clicking on this message. ")],-1),c=9,p=3,d="pa-6",tt=A({__name:"DesktopGettingStartedPage",setup(Be){return(Re,qe)=>(V(),S(h,null,{default:s(()=>[B,R,t(u,{id:"overview",title:"Overview",class:"mb-5","large-title":""},{default:s(()=>[e("div",null,[t(i,{src:a(W),contain:"",eager:"",class:"screenshot ma-6"},null,8,["src"])]),q,z,H,e("p",null,[n(" If you use our Unipept Desktop application, then please cite Verschaffelt et al. (2021) Journal of Proteome Research: "),t(m,{to:"https://academic.oup.com/bioinformatics/article/32/11/1746/1742840?login=false"},{default:s(()=>[n(" doi:10.1093/bioinformatics/btw039 ")]),_:1})])]),_:1}),t(u,{id:"download",title:"Download",class:"mb-5","large-title":""},{default:s(()=>[N,t(l,{dense:""},{default:s(()=>[t(o,{sm:"12",md:"4"},{default:s(()=>[t(f,{style:{width:"100%"},color:"primary","prepend-icon":"mdi-microsoft-windows",href:"https://github.com/unipept/unipept-desktop/releases/latest/download/Unipept-Desktop-x64.exe"},{default:s(()=>[n(" Download for Windows ")]),_:1})]),_:1}),t(o,{sm:"12",md:"4"},{default:s(()=>[t(T,{"offset-y":""},{activator:s(({props:j})=>[t(f,C({style:{width:"100%"},color:"primary"},j,{"prepend-icon":"mdi-apple"}),{default:s(()=>[n(" Download for macOS ")]),_:2},1040)]),default:s(()=>[t(O,null,{default:s(()=>[t(_,{href:"https://github.com/unipept/unipept-desktop/releases/latest/download/Unipept-Desktop-x64.dmg",title:"Mac with Intel Processor (x86, AMD64)"}),t(_,{href:"https://github.com/unipept/unipept-desktop/releases/latest/download/Unipept-Desktop-arm64.dmg",title:"Mac with Apple Silicon Processor (ARM)"})]),_:1})]),_:1})]),_:1}),t(o,{sm:"12",md:"4"},{default:s(()=>[t(f,{style:{width:"100%"},color:"primary","prepend-icon":"mdi-linux",href:"https://github.com/unipept/unipept-desktop/releases/latest/download/Unipept-Desktop-x86_64.AppImage"},{default:s(()=>[n(" Download for Linux ")]),_:1})]),_:1})]),_:1}),E,t(w,{color:"warning",variant:"outlined"},{default:s(()=>[Y,n(" If your are planning to use the Unipept Desktop application to build custom targeted protein reference databases (which are required for proteogenomics data analyses), then you also need to make sure that "),F,n(" is installed and working on your system. See "),t(m,{to:"/desktop/configuration#docker"},{default:s(()=>[n("this page")]),_:1}),n(" for more information on how to set up Docker Desktop with the Unipept Desktop application. ")]),_:1})]),_:1}),t(u,{id:"windows_installation",title:"Install for Windows",class:"mb-5","large-title":""},{default:s(()=>[t(h,null,{default:s(()=>[t(l,null,{default:s(()=>[t(o,{cols:c},{default:s(()=>[G,K]),_:1}),t(o,{cols:p,class:r(d)},{default:s(()=>[e("a",{href:a(g)},[t(i,{src:a(g),contain:"",eager:"",class:"screenshot"},null,8,["src"])],8,J)]),_:1})]),_:1}),t(l,null,{default:s(()=>[t(o,{cols:c},{default:s(()=>[Q,X]),_:1}),t(o,{cols:p,class:r(d)},{default:s(()=>[e("a",{href:a(b)},[t(i,{src:a(b),contain:"",eager:"",class:"screenshot"},null,8,["src"])],8,Z)]),_:1})]),_:1}),t(l,null,{default:s(()=>[t(o,{cols:c},{default:s(()=>[$,ee]),_:1}),t(o,{cols:p,class:r(d)},{default:s(()=>[e("a",{href:a(y)},[t(i,{src:a(y),contain:"",eager:"",class:"screenshot"},null,8,["src"])],8,te)]),_:1})]),_:1}),t(l,null,{default:s(()=>[t(o,{cols:c},{default:s(()=>[se,ae]),_:1}),t(o,{cols:p,class:r(d)},{default:s(()=>[e("a",{href:a(k)},[t(i,{src:a(k),contain:"",eager:"",class:"screenshot"},null,8,["src"])],8,oe)]),_:1})]),_:1})]),_:1})]),_:1}),t(u,{id:"macos_installation",title:"Install for macOS",class:"mb-5"},{default:s(()=>[t(h,null,{default:s(()=>[t(l,null,{default:s(()=>[t(o,{cols:c},{default:s(()=>[ne,ie]),_:1}),t(o,{cols:p,class:r(d)},{default:s(()=>[e("a",{href:a(v)},[t(i,{src:a(v),contain:"",eager:"",class:"screenshot"},null,8,["src"])],8,le)]),_:1})]),_:1}),t(l,null,{default:s(()=>[t(o,{cols:c},{default:s(()=>[re,ce]),_:1}),t(o,{cols:p,class:r(d)},{default:s(()=>[e("a",{href:a(I)},[t(i,{src:a(I),contain:"",eager:"",class:"screenshot"},null,8,["src"])],8,pe)]),_:1})]),_:1})]),_:1})]),_:1}),t(u,{id:"linux_installation",title:"Install for Linux",class:"mb-5","large-title":""},{default:s(()=>[t(h,null,{default:s(()=>[t(l,null,{default:s(()=>[t(o,{cols:c},{default:s(()=>[de,ue]),_:1}),t(o,{cols:p,class:r(d)},{default:s(()=>[e("a",{href:a(D)},[t(i,{src:a(D),contain:"",eager:"",class:"screenshot"},null,8,["src"])],8,he)]),_:1})]),_:1}),t(l,null,{default:s(()=>[t(o,{cols:c},{default:s(()=>[fe,me]),_:1}),t(o,{cols:p,class:r(d)},{default:s(()=>[e("a",{href:a(x)},[t(i,{src:a(x),contain:"",eager:"",class:"screenshot"},null,8,["src"])],8,_e)]),_:1})]),_:1}),t(l,null,{default:s(()=>[t(o,{cols:c},{default:s(()=>[we,ge]),_:1}),t(o,{cols:p,class:r(d)},{default:s(()=>[e("a",{href:a(U)},[t(i,{src:a(U),contain:"",eager:"",class:"screenshot"},null,8,["src"])],8,be)]),_:1})]),_:1})]),_:1})]),_:1}),t(u,{id:"first_time_use",title:"First time use",class:"mb-5","large-title":""},{default:s(()=>[ye,ke,ve,e("div",Ie,[t(i,{src:a(M),"max-width":"800",eager:"",cover:!1},null,8,["src"])]),De,xe,t(w,{color:"warning",variant:"outlined"},{default:s(()=>[Ue,n(" Changes made to the demo project will not be saved! The demo project is read-only and all changes will be lost when you close the application. ")]),_:1}),je,Ae,Ve,Se,Ce]),_:1}),t(u,{id:"update",title:"Updating the application",class:"mb-5","large-title":""},{default:s(()=>[e("div",Te,[t(i,{src:a(L),"max-width":"800",cover:!1,eager:"",class:"screenshot ma-6"},null,8,["src"]),Oe]),We,Me,e("div",Le,[t(i,{src:a(P),"max-width":"800",cover:!1,eager:"",class:"screenshot ma-6"},null,8,["src"]),Pe])]),_:1})]),_:1}))}});export{tt as default};
