import{d as C,o as d,c as f,w as e,a as t,V as P,b as a,t as y,r as A,e as T,f as D,F as L,g as s,h as k,m as h,i as x,j as B,k as F,l as H,G as $,n as M,u as g,p as N,q as S}from"./index-fa2d734a.js";import{V as z,a as G,b as R}from"./VCard-4d112d66.js";import{_ as U}from"./_plugin-vue_export-helper-c27b6911.js";import{H as j}from"./HeaderBodyCard-f3b47b61.js";import{V as q}from"./VList-6f91db5c.js";import{V as W}from"./VListItem-99841f34.js";import{V}from"./VChip-d9d67e94.js";import{V as E}from"./VTooltip-b5f40b4d.js";import{V as m,a as I}from"./VRow-781382bd.js";import{V as v}from"./VHover-fb20754c.js";import"./VDivider-7dfa9365.js";import"./VOverlay-a17eb942.js";import"./delay-fab8d36b.js";import"./forwardRefs-e2994a01.js";const O=C({__name:"HomePageCard",props:{title:{},asset:{},hover:{type:Boolean},contain:{type:Boolean,default:!1}},setup(_){return(l,w)=>(d(),f(R,{elevation:l.hover?6:2,class:"equal"},{default:e(()=>[t(P,{height:"300",src:l.asset,alt:l.asset,cover:!l.contain},null,8,["src","alt","cover"]),t(z,{class:"feature-title pa-4"},{default:e(()=>[a(y(l.title),1)]),_:1}),t(G,{class:"mt-4"},{default:e(()=>[A(l.$slots,"default",{},void 0,!0)]),_:3})]),_:3},8,["elevation"]))}});const b=U(O,[["__scopeId","data-v-7886160f"]]),Y={class:"d-flex flex-row"},J={class:"flex-grow-1 d-flex align-center"},K={class:"d-flex align-center"},Q=C({__name:"ReleaseOverviewCard",props:{services:{}},setup(_){const l=u=>{const c=new Date(u);return`${c.getDate()} ${c.toLocaleString("en",{month:"long"})} ${c.getFullYear()}`},w=(u,c)=>{const n=new Date(u);return Math.round((Date.now()-n.getTime())/864e5)<c};return(u,c)=>(d(),f(j,{title:"Latest versions"},{default:e(()=>[t(q,null,{default:e(()=>[(d(!0),T(L,null,D(u.services,n=>(d(),f(W,{key:n.name,class:"px-0 px-sm-5",to:n.to,link:"","prepend-icon":n.icon,"append-icon":"mdi-chevron-right"},{default:e(()=>[s("div",Y,[s("div",J,y(n.name),1),s("div",K,[w(n.date,90)?(d(),f(V,{key:0,class:"recent-release mr-2 justify-left",size:"small",label:"",color:"green","prepend-icon":"mdi-flag"},{default:e(()=>[a(" Updated ")]),_:1})):k("",!0),n.version?(d(),f(E,{key:1,text:l(n.date)},{activator:e(({props:r})=>[t(V,h({class:"justify-left",size:"small",label:"",color:"primary"},r),{default:e(()=>[a(y(n.version),1)]),_:2},1040)]),_:2},1032,["text"])):k("",!0)])])]),_:2},1032,["to","prepend-icon"]))),128))]),_:1})]),_:1}))}}),X="/unipept/assets/home-feature-single-5100388f.png",Z="/unipept/assets/home-feature-multi-9d66fb89.svg",ee="/unipept/assets/home-feature-desktop-bc27bb3a.png",te="/unipept/assets/home-feature-api-9b0094c5.svg",ae="/unipept/assets/home-feature-cli-4001b2f6.svg",se="/unipept/assets/home-feature-umgap-3c398674.svg",o=_=>(N("data-v-22c29452"),_=_(),S(),_),oe=o(()=>s("div",{class:"text-h3 font-weight-light mb-2"}," Welcome ",-1)),ne=o(()=>s("a",{href:"https://www.ugent.be/en",target:"_blank"},"Ghent University",-1)),ie=o(()=>s("span",{class:"font-weight-bold"},"interactive datavisualizations",-1)),le=o(()=>s("span",{class:"font-weight-bold"},"biodiversity analysis",-1)),re=o(()=>s("span",{class:"font-weight-bold"},"metaproteome samples",-1)),pe=o(()=>s("span",{class:"font-weight-bold"},"targeted proteomics",-1)),ce=o(()=>s("span",{class:"font-weight-bold"},"comparing genomes",-1)),de={key:1,class:"d-flex justify-center"},ue=o(()=>s("span",{class:"font-weight-bold"},"tryptic peptide",-1)),me=o(()=>s("span",{class:"font-weight-bold"},"all Uniprot entries",-1)),he=o(()=>s("span",{class:"font-weight-bold"},"lists of tryptic peptides",-1)),fe=o(()=>s("span",{class:"font-weight-bold"},"biodiversity of your sample",-1)),_e=o(()=>s("span",{class:"font-weight-bold"},"web service",-1)),ge=o(()=>s("span",{class:"font-weight-bold"},"documentation pages",-1)),ve=o(()=>s("span",{class:"font-weight-bold"},"command line interface",-1)),be=o(()=>s("span",{class:"font-weight-bold"},"integrate Unipept",-1)),we=o(()=>s("span",{class:"font-weight-bold"},"pipelines and scripts",-1)),ye=o(()=>s("span",{class:"font-weight-bold"},"shotgun metagenomics reads",-1)),Ce=C({__name:"HomePage",setup(_){const l=new $,w=x(!0),u=x([]);return B(async()=>{const c=await l.latestRelease("https://api.github.com/repos/unipept/unipept-api/releases"),n=await l.latestRelease("https://api.github.com/repos/unipept/unipept-cli/releases"),r=await l.latestRelease("https://api.github.com/repos/unipept/unipept/releases"),i=await l.latestRelease("https://api.github.com/repos/unipept/unipept-desktop/releases");u.value=[{name:"API",icon:"mdi-api",version:c.tag_name.replace(/^v/,""),date:c.published_at,to:"/news/api"},{name:"CLI",icon:"mdi-console",version:n.tag_name.replace(/^v/,""),date:n.published_at,to:"/news/cli"},{name:"Web app",icon:"mdi-web",version:r.tag_name.replace(/^v/,""),date:r.published_at,to:"/news/web"},{name:"Desktop app",icon:"mdi-desktop-tower-monitor",version:i.tag_name.replace(/^v/,""),date:i.published_at,to:"/news/desktop"}],w.value=!1}),(c,n)=>{const r=F("router-link");return d(),f(H,null,{default:e(()=>[t(I,null,{default:e(()=>[t(m,{cols:"12",md:"6"},{default:e(()=>[oe,a(" Unipept is an open source web application developed at "),ne,a(" that is designed for metaproteomics data analysis with a focus on "),ie,a(". Unipept is powered by an index containing all UniProt entries, a tweaked version of the NCBI taxonomy and a custom lowest common ancestor algorithm. This combination enables a blazingly fast "),le,a(" of large and complex "),re,a(". This functionality is also available via an API and a set of command line tools. Next to these core functions, Unipept also has a tool for selecting unique peptides for "),pe,a(" and for "),ce,a(" based on peptide similarity. ")]),_:1}),t(m,{class:"mt-0",cols:"12",md:"6"},{default:e(()=>[w.value?(d(),T("div",de,[t(M,{indeterminate:"",color:"primary"})])):(d(),f(Q,{key:0,services:u.value},null,8,["services"]))]),_:1})]),_:1}),t(I,null,{default:e(()=>[t(m,{sm:"6",lg:"4"},{default:e(()=>[t(r,{to:"/tpa"},{default:e(()=>[t(v,null,{default:e(({isHovering:i,props:p})=>[t(b,h({hover:i||!1,title:"Tryptic Peptide Analysis",asset:g(X)},p),{default:e(()=>[a(" With tryptic peptide analysis, you can submit a single "),ue,a(" that can be 5 to 50 residues long. The application will respond with a list of "),me,a(" wherein the peptide was found along with a complete taxonomic lineage derived from the NCBI taxonomy. These lineages are presented as a comprehensible table and using an interactive tree view. ")]),_:2},1040,["hover","asset"])]),_:1})]),_:1})]),_:1}),t(m,{sm:"6",lg:"4"},{default:e(()=>[t(r,{to:"/mpa"},{default:e(()=>[t(v,null,{default:e(({isHovering:i,props:p})=>[t(b,h({hover:i||!1,title:"Metaproteomics Analysis",asset:g(Z)},p),{default:e(()=>[a(" Metaproteomics analysis helps you analyze "),he,a(", e.g. extracted from an environmental sample using shotgun tandem mass spectrometric methods. Of these peptides, the lowest common ancestors (LCA) will be calculated. These LCAs will be bundled and displayed in an interactive treemap giving you insight into the "),fe,a(". ")]),_:2},1040,["hover","asset"])]),_:1})]),_:1})]),_:1}),t(m,{sm:"6",lg:"4"},{default:e(()=>[t(r,{to:"/desktop"},{default:e(()=>[t(v,null,{default:e(({isHovering:i,props:p})=>[t(b,h({hover:i||!1,title:"Unipept Desktop",asset:g(ee)},p),{default:e(()=>[a(" The Unipept Desktop application allows to analyse larger samples, to store analysis results offline, to extensively compare assays with each other and to better manage samples. This application is powered by the Electron framework and runs on almost all systems. This allows you to improve the understanding of all interactions that are taking place in a complex environment. ")]),_:2},1040,["hover","asset"])]),_:1})]),_:1})]),_:1}),t(m,{sm:"6",lg:"4"},{default:e(()=>[t(r,{to:"/apidocs"},{default:e(()=>[t(v,null,{default:e(({isHovering:i,props:p})=>[t(b,h({hover:i||!1,title:"API Documentation",asset:g(te),contain:""},p),{default:e(()=>[a(" Unipept offers most of its peptide analysis features as a "),_e,a(". This enables the integration of Unipept functionality into other applications and the creation of batch processing scripts. These "),ge,a(" describe the available features of the API, how to access them and plenty examples. ")]),_:2},1040,["hover","asset"])]),_:1})]),_:1})]),_:1}),t(m,{sm:"6",lg:"4"},{default:e(()=>[t(r,{to:"/clidocs"},{default:e(()=>[t(v,null,{default:e(({isHovering:i,props:p})=>[t(b,h({hover:i||!1,title:"CLI Documentation",asset:g(ae),contain:""},p),{default:e(()=>[a(" The Unipept "),ve,a(" (CLI) is a wrapper around the Unipept API that offers an easy way to "),be,a(" functionality into your data processing "),we,a(". These pages cover installation and usage instructions, an overview of all available options and several use cases. ")]),_:2},1040,["hover","asset"])]),_:1})]),_:1})]),_:1}),t(m,{sm:"6",lg:"4"},{default:e(()=>[t(r,{to:"/umgap"},{default:e(()=>[t(v,null,{default:e(({isHovering:i,props:p})=>[t(b,h({hover:i||!1,title:"UMGAP",asset:g(se),contain:""},p),{default:e(()=>[a(" Use the Unipept MetaGenomics Analysis Pipeline to assign taxonomic labels to your "),ye,a(". The results are available as taxonomic frequency tables and interactive visualizations. UMGAP is a collection of CLI tools that can be combined to identify shotgun metagenomics reads. ")]),_:2},1040,["hover","asset"])]),_:1})]),_:1})]),_:1})]),_:1})]),_:1})}}});const Me=U(Ce,[["__scopeId","data-v-22c29452"]]);export{Me as default};
