import{I as i}from"./InlineCode-b941903d.js";import{H as u}from"./HeaderBodyCard-f3b47b61.js";import{I as o}from"./Initialism-25eec5f3.js";import{B as s,S as a}from"./Boxed-1ca540a2.js";import{_ as d}from"./ParameterTable.vue_vue_type_script_setup_true_lang-ea4e3ff2.js";import{d as c,o as m,c as p,w as l,l as h,g as t,b as e,a as n}from"./index-fa2d734a.js";import{V as _}from"./VDivider-7dfa9365.js";import"./_plugin-vue_export-helper-c27b6911.js";import"./VCard-4d112d66.js";import"./VRow-781382bd.js";const f=t("h1",{class:"font-weight-light"}," umgap pept2lca ",-1),g={class:"font-weight-light"},b=t("i",null,"standard input",-1),y=t("b",null,"cat input.fa",-1),x=t("br",null,null,-1),A=t("br",null,null,-1),T=t("br",null,null,-1),k=t("br",null,null,-1),w=t("b",null,"umgap pept2lca tryptic-peptides.index < input.fa",-1),B=t("br",null,null,-1),I=t("br",null,null,-1),P=t("br",null,null,-1),V=t("b",null,"cat input.fa",-1),N=t("br",null,null,-1),C=t("br",null,null,-1),q=t("br",null,null,-1),v=t("br",null,null,-1),F=t("b",null,"umgap pept2lca -o tryptic-peptides.index < input.fa",-1),S=t("br",null,null,-1),E=t("br",null,null,-1),L=t("br",null,null,-1),Q=c({__name:"MetagenomicsPept2LcaPage",setup($){const r=[{name:"-h / --help",description:"Prints help information"},{name:"-V / --version",description:"Prints version information"},{name:"-m / --in-memory",description:"Load index in memory instead of memory mapping the file contents. This makes querying significantly faster, but requires some initialization time."},{name:"-o / --one-on-one",description:"Map unknown sequences to 0 instead of ignoring them"},{name:"-c / --chunksize c",description:"Number of reads grouped into one chunk. Bigger chunks decrease the overhead caused by multithreading. Because the output order is not necessarily the same as the input order, having a chunk size which is a multiple of 12 (all 6 translations multiplied by the two paired-end reads) will keep FASTA records that originate from the same reads together [default: 240]"}];return(D,z)=>(m(),p(h,null,{default:l(()=>[f,t("h3",g,[e("Maps a "),n(o,null,{default:l(()=>[e("FASTA")]),_:1}),e(" stream of peptides to taxon IDs.")]),n(_,{class:"my-2"}),t("p",null,[e(" The "),n(i,null,{default:l(()=>[e("umgap pept2lca")]),_:1}),e(" command takes one or more amino acid sequences and looks up the corresponding taxon ID in an index file (as build by the "),n(i,null,{default:l(()=>[e("umgap buildindex")]),_:1}),e(" command). ")]),n(u,{id:"usage",title:"Usage","large-title":""},{default:l(()=>[t("p",null,[e(" The input is given in "),n(o,null,{default:l(()=>[e("FASTA")]),_:1}),e(" format on "),b,e(". Per "),n(o,null,{default:l(()=>[e("FASTA")]),_:1}),e(" header, there can be multiple sequences, each on a line. In the following example we match tryptic peptides on their lowest common ancestor in the "),n(o,null,{default:l(()=>[e("NCBI")]),_:1}),e(" taxonomy. ")]),n(s,{style:{"max-height":"fit-content"}},{default:l(()=>[n(a,null,{default:l(()=>[e("$")]),_:1}),e(),y,x,e(">header1 "),A,e("AAALTER "),T,e("ENFVYLAK "),k,n(a,null,{default:l(()=>[e("$")]),_:1}),e(),w,B,e(">header1 "),I,e("2 "),P,e("3398 ")]),_:1}),t("p",null,[e(" By default, sequences not found in the index are ignored. Using the "),n(i,null,{default:l(()=>[e("-o (--on-on-one)")]),_:1}),e(" flag, they are mapped to 0, instead. ")]),n(s,null,{default:l(()=>[n(a,null,{default:l(()=>[e("$")]),_:1}),e(),V,N,e(">header1 "),C,e("NOTATRYPTICPEPTIDE "),q,e("ENFVYLAK "),v,n(a,null,{default:l(()=>[e("$")]),_:1}),e(),F,S,e(">header1 "),E,e("0 "),L,e("3398 ")]),_:1}),n(d,{class:"mt-5 mb-3",parameters:r})]),_:1})]),_:1}))}});export{Q as default};
