import{I as i}from"./InlineCode-67d03729.js";import{H as r}from"./HeaderBodyCard-3295d4c5.js";import{I as o}from"./Initialism-440d96b0.js";import{B as d,S as s}from"./Boxed-9f9dd66a.js";import{_ as p}from"./ParameterTable.vue_vue_type_script_setup_true_lang-eb31a2e0.js";import{d as m,o as u,c,w as a,l as h,g as t,b as e,a as n}from"./index-eea28a33.js";import{V as f}from"./VDivider-834b9f8e.js";import"./_plugin-vue_export-helper-c27b6911.js";import"./VCard-9d59c194.js";import"./VRow-94e3f3e6.js";const _=t("h1",{class:"font-weight-light"}," umgap prot2tryp2lca ",-1),g={class:"font-weight-light"},y=t("i",null,"standard input",-1),b=t("i",null,"standard output",-1),w=t("b",null,"cat input.fa",-1),T=t("br",null,null,-1),A=t("br",null,null,-1),V=t("br",null,null,-1),k=t("b",null,"umgap prot2tryp2lca tryptic-lca.index < input.fa",-1),x=t("br",null,null,-1),D=t("br",null,null,-1),I=t("br",null,null,-1),N=t("br",null,null,-1),v=t("br",null,null,-1),H=m({__name:"MetagenomicsProt2Tryp2LcaPage",setup(L){const l=[{name:"-h / --help",description:"Prints help information"},{name:"-V / --version",description:"Prints version information"},{name:"-m / --in-memory",description:"Load index in memory instead of memory mapping the file contents. This makes querying significantly faster, but requires some initialization time."},{name:"-o / --one-on-one",description:"Map unknown sequences to 0 instead of ignoring them"},{name:"-c / --chunksize c",description:"Number of reads grouped into one chunk. Bigger chunks decrease the overhead caused by multithreading. Because the output order is not necessarily the same as the input order, having a chunk size which is a multiple of 12 (all 6 translations multiplied by the two paired-end reads) will keep FASTA records that originate from the same reads together [default: 240]"},{name:"-k / --keep k",description:"Amino acid symbols that a sequence must contain [none by default]"},{name:"-d / --drop d",description:"Amino acid symbols that a sequence may not contain [none by default]"},{name:"-L / --maxlen L",description:"Maximum length allowed [default: 50]"},{name:"-l / --minlen l",description:"Minimum length required [default: 5]"},{name:"-p / --pattern p",description:"The cleavage-pattern (regex), i.e. the pattern after which the next peptide will be cleaved for tryptic peptides) [default: ([KR])([^P])]"}];return(S,B)=>(u(),c(h,null,{default:a(()=>[_,t("h3",g,[e(" Digests a "),n(o,null,{default:a(()=>[e("FASTA")]),_:1}),e(" stream of peptides and maps all tryptic peptides to taxon IDs. ")]),n(f,{class:"my-2"}),t("p",null,[e(" The "),n(i,null,{default:a(()=>[e("umgap prot2tryp2lca")]),_:1}),e(" command takes one or more peptides and splits these into tryptic peptides, possibly filters them, and outputs their lowest common ancestors. It is a combination of the "),n(i,null,{default:a(()=>[e("umgap prot2tryp")]),_:1}),e(", "),n(i,null,{default:a(()=>[e("umgap filter")]),_:1}),e(" and "),n(i,null,{default:a(()=>[e("umgap pept2lca")]),_:1}),e(" commands to allow more efficient parallel computing (c.f. their documentation for details). ")]),n(r,{id:"usage",title:"Usage","large-title":""},{default:a(()=>[t("p",null,[e(" The input is given in a "),n(o,null,{default:a(()=>[e("FASTA")]),_:1}),e(" format on "),y,e(" with a single peptide per "),n(o,null,{default:a(()=>[e("FASTA")]),_:1}),e(" header, which may be hardwrapped with newlines. The command prints the lowest common ancestors for each tryptic peptide found in each given peptide to "),b,e(". ")]),n(d,{style:{"max-height":"fit-content"}},{default:a(()=>[n(s,null,{default:a(()=>[e("$")]),_:1}),e(),w,T,e(">header1 "),A,e("AYKKAGVSGHVWQSDGITNCLLRGLTRVKEAVANRDSGNGYINKVYYWTVDKRATTRDALDAGVDGIMTNYPDVITDVLN "),V,n(s,null,{default:a(()=>[e("$")]),_:1}),e(),k,x,e(">header1 "),D,e("571525 "),I,e("1 "),N,e("571525 "),v,e("6920 ")]),_:1}),n(p,{class:"mt-5 mb-3",parameters:l})]),_:1})]),_:1}))}});export{H as default};
