import{H as u}from"./HeaderBodyCard-3295d4c5.js";import{R as d}from"./ResourceLink-d34b4283.js";import{B as c,S as o}from"./Boxed-9f9dd66a.js";import{I as s}from"./InlineCode-67d03729.js";import{I as l}from"./Initialism-440d96b0.js";import{_ as p}from"./ParameterTable.vue_vue_type_script_setup_true_lang-eb31a2e0.js";import{_ as G}from"./ImageCaptionCard.vue_vue_type_script_setup_true_lang-dcecb080.js";import{d as w,o as m,c as A,w as i,l as C,g as t,b as e,a as n,t as y,u as v,e as T,f as S,F as P,p as k,q as I}from"./index-eea28a33.js";import{_ as x}from"./_plugin-vue_export-helper-c27b6911.js";import"./VCard-9d59c194.js";import"./VRow-94e3f3e6.js";const z="/assets/casestudy-umgap-basic-8ef71676.png",a=r=>(k("data-v-d134efe9"),r=r(),I(),r),M=a(()=>t("h1",{class:"font-weight-light"}," Case Study: Running a preconfigured metagenomics pipeline ",-1)),F={class:"font-weight-light"},E=a(()=>t("p",null," The following pipelines are preconfigured: ",-1)),q=a(()=>t("p",null," Following code snippet shows the interaction with the setup script. It starts out by asking the relevant questions, without delay. At the end, it downloads the relevant files in sequence, without further interaction. ",-1)),B=a(()=>t("br",null,null,-1)),U=a(()=>t("br",null,null,-1)),D=a(()=>t("br",null,null,-1)),$=a(()=>t("br",null,null,-1)),H=a(()=>t("br",null,null,-1)),L=a(()=>t("br",null,null,-1)),K=a(()=>t("br",null,null,-1)),V=a(()=>t("br",null,null,-1)),J=a(()=>t("p",null," It seems the pipeline cannot be more specific than root (Taxon ID 1) for the first read. ",-1)),R=a(()=>t("br",null,null,-1)),W=a(()=>t("br",null,null,-1)),O=a(()=>t("br",null,null,-1)),N=a(()=>t("br",null,null,-1)),Q=a(()=>t("br",null,null,-1)),X=a(()=>t("br",null,null,-1)),Z=a(()=>t("br",null,null,-1)),j=a(()=>t("br",null,null,-1)),Y=a(()=>t("br",null,null,-1)),ee=a(()=>t("h2",{class:"font-weight-light mt-5"}," References ",-1)),te=w({__name:"MetagenomicsPreconfiguredPage",setup(r){const _=[{name:"-c dir",description:"Set a different configuration directory."},{name:"-f dir",description:"Link the location of your FragGeneScan installation."},{name:"-d dir",description:"Set a different location for downloading the database files. This can be on a separate disk."},{name:"-y",description:"Download all files of the latest version without asking for confirmation."}],g=[{name:"-c dir",description:"Set a different configuration directory."},{name:"-z",description:"Request compression of the output file."},{name:"-1 file",description:"Single ended FASTA or first pair-ended FASTQ input file, optionally compressed."},{name:"-2 file",description:"Second pair-ended FASTQ input file, optionally compressed."},{name:"-t",description:"Type of the analysis (high precision by default)."},{name:"-o file",description:"The output file."}],b=[{name:"-t",description:"Output a CSV frequency table on species rank."},{name:"-w",description:"Output an HTML webpage of an interactive visualization."},{name:"-u",description:"Print a shareable URL to a online interactive visualisation."},{name:"-r rank",description:"Set the rank for the CSV frequency table (default: species)."}],h={lindgreen:{full:"Lindgreen, S., Adair, K. L., & Gardner, P. P. (2016). An evaluation of the accuracy and speed of metagenome analysis tools. Scientific reports, 6, 19233.",short:"Lindgreen et al. (2016)"}};return(ne,ie)=>(m(),A(C,null,{default:i(()=>[M,t("h3",F,[e(" This case study shows how to set up and run the preconfigured scripts of the "),n(l,null,{default:i(()=>[e("UMGAP")]),_:1}),e(" for the taxonomic analysis of a metagenomics dataset. ")]),n(u,{class:"mt-5",title:"Introduction","large-title":""},{default:i(()=>[t("p",null,[e(" The Unipept Metagenomics Analysis Pipeline ("),n(l,null,{default:i(()=>[e("UMGAP")]),_:1}),e(") is accompanied by 3 scripts: "),n(s,null,{default:i(()=>[e("umgap-setup.sh")]),_:1}),e(", "),n(s,null,{default:i(()=>[e("umgap-analyse.sh")]),_:1}),e(" and "),n(s,null,{default:i(()=>[e("umgap-visualize.sh")]),_:1}),e(". The setup script deals with downloading prebuilt databases and checking and linking external tools. The analyse script runs several analyses in sequence. The visualize script creates webpages from the results of the analyse script. ")]),E,t("ul",null,[t("li",null,[t("b",null,[n(l,null,{default:i(()=>[e("UMGAP")]),_:1}),e(" High Precision")]),e(": Optimized for high precision identifications on your metagenomics reads.")]),t("li",null,[t("b",null,[n(l,null,{default:i(()=>[e("UMGAP")]),_:1}),e(" Max Precision")]),e(": Optimized for highest precision, with a small setback on sensitivity.")]),t("li",null,[t("b",null,[n(l,null,{default:i(()=>[e("UMGAP")]),_:1}),e(" Tryptic Precision")]),e(": Made for fast analyses on your laptop. Fewer results, but accurate.")]),t("li",null,[t("b",null,[n(l,null,{default:i(()=>[e("UMGAP")]),_:1}),e(" High Sensitivity")]),e(": Optimized for high sensitivity identifications on metagenomics reads.")]),t("li",null,[t("b",null,[n(l,null,{default:i(()=>[e("UMGAP")]),_:1}),e(" Max Sensitivity")]),e(": Optimized for highest sensitivity, with a small setback on precision.")]),t("li",null,[t("b",null,[n(l,null,{default:i(()=>[e("UMGAP")]),_:1}),e(" Tryptic Sensitivity")]),e(": Made for fast analyses on your laptop. Get a quick overview of your data.")])])]),_:1}),n(u,{class:"mt-5",title:"Setup","large-title":""},{default:i(()=>[t("p",null,[e(" The preconfigured pipelines require some databases and external tools. The setup script will help you to get these in the right place. In general, you only need to run this script once, but you can also use it to verify the file locations and download newer versions of the data. If you're planning to use the tryptic precision, tryptic sensitivity, high precision or max precision pipelines, you'll need to install "),n(d,{to:"https://github.com/unipept/FragGeneScanPlusPlus/"},{default:i(()=>[e("FragGeneScan++")]),_:1}),e(" first. In this casestudy, we will use a tryptic pipeline, so we don't have to download the 9-mer index (~12GiB, rather than ~150GiB). ")]),q,n(c,null,{default:i(()=>[t("pre",null,[e(""),n(o,null,{default:i(()=>[e("$")]),_:1}),e(` umgap-setup.sh -f /opt/FragGeneScanPlusPlus
Use '/home/user/.config/unipept' as configuration directory? [y]/n y
Created directory /home/user/.config/unipept.
Found, tested and remembered the FragGeneScan++ location.
Use '/home/user/.local/share/unipept' as data directory? [y]/n y
Created directory /home/user/.local/share/unipept.
Checking the latest version on the server.
Latest version is 2020-12-02.

For any type of analysis, you need a taxonomony. For mapping tryptic or
9-mer peptides to it, you need the respective index file of the same
version.

Would you like to download the taxonomy from 2020-12-02 (115MiB)? [y]/n y
Would you like to download the tryptic index from 2020-12-02 (12GiB)? [y]/n y
Would you like to download the 9-mer index from 2020-12-02 (152GiB)? [y]/n n
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  114M  100  114M    0     0  6477k      0  0:00:18  0:00:18 --:--:-- 6332k
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   12G  100   12G    0     0  5076k      0  0:34:12  0:34:12 --:--:-- 5790k
`)])]),_:1}),t("p",null,[e(" To change where the setup script will store the configuration and downloaded files, add some of the options detailed below. If you're familiar with the setup script, or you're running the script within another script, you can add the "),n(s,null,{default:i(()=>[e("-y")]),_:1}),e(" flag to run without interaction. ")]),n(p,{class:"my-3",parameters:_})]),_:1}),n(u,{class:"mt-5",title:"Analysis","large-title":""},{default:i(()=>[t("p",null,[e(" When setup is complete, the analyse script will automatically use the downloaded databases and linked tools. The input data consists of a 100 paired-end reads sampled from a dataset generated by "+y(h.lindgreen.short)+" for their metabenchmark. With these paired-end reads in the ",1),n(d,{to:"/A1.fq",target:"_blank"},{default:i(()=>[e("A1.fq")]),_:1}),e(" and "),n(d,{to:"/A2.fq",target:"_blank"},{default:i(()=>[e("A2.fq")]),_:1}),e(" files, the commands are as follows. ")]),n(c,null,{default:i(()=>[n(o,null,{default:i(()=>[e("$")]),_:1}),e(" head -4 A1.fq "),B,e("@1198114###CP002480-_Acidobacteria_733918/1 "),U,e("ATCGCGCACGCGGCCGATGCCCCAGAAGAGATTGACAGCGGTGGGGCGGGCGGCGGCGAGGTGGTCGCAGATCTCGGCGACCTCTGCGTTGAGGGTCGGG "),D,e("+ "),$,e("AADEGGAGIIEIHFHKHJKKJKHIJJIIKCJAHBFKKFHIJIF;JIH5DE$I>CBE$E:FGEJCGABEEECCDKD?D?C$ECEEE;CEEDEEAD=?ED$D "),H,n(o,null,{default:i(()=>[e("$")]),_:1}),e(" umgap-analyse.sh -1 A1.fq -2 A2.fq -t tryptic-precision -z -o tryptic-precision-output.fa.gz "),L,n(o,null,{default:i(()=>[e("$")]),_:1}),e(" zcat tryptic-precision-output.fa.gz | head -2 "),K,e(">1198114###CP002480-_Acidobacteria_733918/1_1_100_- "),V,e("1 ")]),_:1}),J,t("p",null,[e(" If you used a non-default configuration directory for setup, you'll need to pass the same directory here, with the "),n(s,null,{default:i(()=>[e("-c")]),_:1}),e(" option. When running the pipeline on large samples, it may be useful to compress the output. Use the "),n(s,null,{default:i(()=>[e("-z")]),_:1}),e(" flag to "),n(l,null,{default:i(()=>[e("GZIP")]),_:1}),e(" the results. Should the input be "),n(l,null,{default:i(()=>[e("GZIP")]),_:1}),e("-compressed, the script will detect this. Note that all arguments can be repeated to bundle multiple analysis, each series ending with a "),n(s,null,{default:i(()=>[e("-o")]),_:1}),e(" option. ")]),n(p,{class:"my-3",parameters:g})]),_:1}),n(u,{class:"mt-5",title:"Visualize","large-title":""},{default:i(()=>[t("p",null,[e(" After analysis, you probably want to view the results. While the analyse script output format is easily parsed, you can also use the visualize script to create importable "),n(l,null,{default:i(()=>[e("CSV")]),_:1}),e(" frequency tables and interactive visualisations. The interactive visualisations can be stored locally or hosted online. The following snippet creates all three in turn. ")]),n(c,null,{default:i(()=>[n(o,null,{default:i(()=>[e("$")]),_:1}),e(" umgap-visualize.sh -t -r phylum tryptic-precision-output.fa.gz "),R,e("taxon id,taxon name,tryptic-precision-output.fa.gz "),W,e("1117,Cyanobacteria,2 "),O,e("57723,Acidobacteria,4 "),N,e("201174,Actinobacteria,4 "),Q,e("1224,Proteobacteria,5 "),X,e("1,root,160 "),Z,n(o,null,{default:i(()=>[e("$")]),_:1}),e(" umgap-visualize.sh -w tryptic-sensitivity-output.fa.gz > tryptic-sensitivity.html "),j,n(o,null,{default:i(()=>[e("$")]),_:1}),e(" umgap-visualize.sh -u tryptic-sensitivity-output.fa.gz "),Y,e("tryptic-sensitivity-output.fa.gz: "),n(d,{to:"https://bl.ocks.org/11b7809d6754b9530cf1a49d93a8d568"},{default:i(()=>[e("https://bl.ocks.org/11b7809d6754b9530cf1a49d93a8d568")]),_:1})]),_:1}),t("p",null,[e(" The "),n(l,null,{default:i(()=>[e("CSV")]),_:1}),e(" tables contain a record for each taxon found in the sample. A record contains, in order, the taxon ID, (for convenience) the taxon name and the number of reads assigned to this taxon or below. All records will be at the same specified taxon rank or at root (unidentified). ")]),n(p,{class:"my-3",parameters:b}),n(G,{class:"mt-5",image:v(z)},null,8,["image"])]),_:1}),ee,t("ul",null,[(m(),T(P,null,S(h,f=>t("li",{key:f.short},y(f.full),1)),64))])]),_:1}))}});const me=x(te,[["__scopeId","data-v-d134efe9"]]);export{me as default};
