import{H as l}from"./HeaderBodyCard-f3b47b61.js";import{R as m}from"./ResourceLink-82ddf6bf.js";import{u as p}from"./useNavigation-4732dd9e.js";import{I as s}from"./Initialism-25eec5f3.js";import{d as f,o as h,c as g,w as a,l as _,g as i,a as t,b as e,K as r,u as c}from"./index-fa2d734a.js";import{V as u}from"./VDivider-7dfa9365.js";import"./VCard-4d112d66.js";import"./_plugin-vue_export-helper-c27b6911.js";const v={class:"font-weight-light"},y=i("h3",{class:"font-weight-light"}," This page explores several uses of the Unipept MetaGenomics Analysis Pipeline. ",-1),w={class:"mb-0"},b={class:"mb-0"},x={class:"mb-0"},I=f({__name:"MetagenomicsCaseStudiesPage",setup(P){const{navigate:n}=p();return(k,o)=>(h(),g(_,null,{default:a(()=>[i("h1",v,[t(s,null,{default:a(()=>[e("UMGAP")]),_:1}),e(" Case Studies ")]),y,t(l,{class:"mt-5",title:"Introduction","large-title":""},{default:a(()=>[i("p",null,[e(" The Unipept Metagenomics Analysis Pipeline ("),t(s,null,{default:a(()=>[e("UMGAP")]),_:1}),e(") is a tool for mapping (short) metagenomics reads to taxa, rivalling state of the art alternatives such as Kraken (2014) with the additional benefit of species level identification. "),t(s,null,{default:a(()=>[e("UMGAP")]),_:1}),e(" achieves this by transforming the mapping task into a metaproteomics problem by using various alternative gene predictors on the metagenomics reads and processing the resulting data with the existing Unipept Metaproteomics Analysis Pipeline. Each step in "),t(s,null,{default:a(()=>[e("UMGAP")]),_:1}),e(" is highly configurable. ")]),i("p",null,[e(" These case studies assume you have already followed the installation instructions on the "),t(m,{to:"/umgap",router:""},{default:a(()=>[e("overview")]),_:1}),e(" page. ")])]),_:1}),t(l,{class:"mt-5",title:"Preconfigured analysis of a metagenomics dataset","large-title":""},{default:a(()=>[i("p",w,[e(" The easiest way to use "),t(s,null,{default:a(()=>[e("UMGAP")]),_:1}),e(" for a taxonomic analysis of a metagenomic dataset is with one of the 6 preconfigured pipelines. This case study walks you through the complete analysis of 2 short "),t(s,null,{default:a(()=>[e("FASTQ")]),_:1}),e(" files. ")]),t(u,{class:"my-2"}),t(r,{color:"secondary",class:"float-end mb-2",variant:"text",onClick:o[0]||(o[0]=d=>c(n)("/umgap/casestudies/basic"))},{default:a(()=>[e(" Get started ")]),_:1})]),_:1}),t(l,{class:"mt-5",title:"Complete configuration walkthrough","large-title":""},{default:a(()=>[i("p",b,[e(" This case study shows how to use "),t(s,null,{default:a(()=>[e("UMGAP")]),_:1}),e(" for the taxonomic analysis of a metagenomics dataset. It shows every step of the pipeline in detail, refering to the documentation for alternative options. ")]),t(u,{class:"my-2"}),t(r,{color:"secondary",variant:"text",class:"float-end mb-2",onClick:o[1]||(o[1]=d=>c(n)("/umgap/casestudies/advanced"))},{default:a(()=>[e(" Get started ")]),_:1})]),_:1}),t(l,{class:"mt-5",title:"Comparative analysis walkthrough","large-title":""},{default:a(()=>[i("p",x,[e(" This case study shows how to use "),t(s,null,{default:a(()=>[e("UMGAP")]),_:1}),e(" for the comparative taxonomic analysis of multiple metagenomics datasets. Using a preconfigured pipeline and the visualize script, we show how to create a comparative graphic. ")]),t(u,{class:"my-2"}),t(r,{color:"secondary",class:"float-end mb-2",variant:"text",onClick:o[2]||(o[2]=d=>c(n)("/umgap/casestudies/comparative"))},{default:a(()=>[e(" Get started ")]),_:1})]),_:1})]),_:1}))}});export{I as default};
