import{I as a}from"./InlineCode-b941903d.js";import{H as r}from"./HeaderBodyCard-f3b47b61.js";import{I as o}from"./Initialism-25eec5f3.js";import{B as u,S as s}from"./Boxed-1ca540a2.js";import{_ as d}from"./ParameterTable.vue_vue_type_script_setup_true_lang-ea4e3ff2.js";import{d as _,o as h,c,w as l,l as f,a as n,g as t,b as e}from"./index-fa2d734a.js";import{V as m}from"./VDivider-7dfa9365.js";import"./_plugin-vue_export-helper-c27b6911.js";import"./VCard-4d112d66.js";import"./VRow-781382bd.js";const p=t("h1",{class:"font-weight-light"}," umgap bestof ",-1),b=t("h3",{class:"font-weight-light"}," Selects the best read of every fixed size group ",-1),C=t("i",null,"standard input",-1),A=t("i",null,"standard output",-1),G=t("b",null,"cat dna.fa",-1),g=t("br",null,null,-1),T=t("br",null,null,-1),x=t("br",null,null,-1),w=t("b",null,"umgap translate -n -a < dna.fa | umgap prot2kmer2lca 9mer.index | tee input.fa",-1),I=t("br",null,null,-1),y=t("br",null,null,-1),B=t("br",null,null,-1),V=t("br",null,null,-1),k=t("br",null,null,-1),v=t("br",null,null,-1),D=t("br",null,null,-1),P=t("br",null,null,-1),S=t("br",null,null,-1),$=t("br",null,null,-1),F=t("br",null,null,-1),N=t("br",null,null,-1),R=t("b",null,"umgap bestof < input.fa",-1),H=t("br",null,null,-1),z=t("br",null,null,-1),M=t("p",null," Taxon IDs are separated by newlines in the actual output, but are separated by spaces in this example. ",-1),ee=_({__name:"MetagenomicsBestofPage",setup(U){const i=[{name:"-h / --help",description:"Prints help information"},{name:"-V / --version",description:"Prints version information"},{name:"-f / --frames f",description:"The number of frames of which to pick the best [default: 6]"}];return(j,q)=>(h(),c(f,null,{default:l(()=>[p,b,n(m,{class:"my-2"}),t("p",null,[e(" The "),n(a,null,{default:l(()=>[e("umgap bestof")]),_:1}),e(" command takes groups of taxon IDs as input and outputs for each group the taxon ID with the most non-root identifications. ")]),n(r,{id:"usage",title:"Usage","large-title":""},{default:l(()=>[t("p",null,[e(" The input is given in "),n(o,null,{default:l(()=>[e("FASTA")]),_:1}),e(" format on "),C,e(". Per "),n(o,null,{default:l(()=>[e("FASTA")]),_:1}),e(" header, there should be multiple numbers (taxon IDs). Per 6 "),n(o,null,{default:l(()=>[e("FASTA")]),_:1}),e(" records (or whichever number you specify with "),n(a,null,{default:l(()=>[e("-f")]),_:1}),e("), the best record is selected and written to "),A,e(". If the input is a series of identified taxon IDs for each of the 6 translations of a read, the output will most likely come from the actual coding frame. ")]),n(u,{style:{"max-height":"fit-content"}},{default:l(()=>[n(s,null,{default:l(()=>[e("$")]),_:1}),e(),G,g,e(">header1 "),T,e("CGCAGAGACGGGTAGAACCTCAGTAATCCGAAAAGCCGGGATCGACCGCCCCTTGCTTGCAGCCGGGCACTACAGGACCC "),x,n(s,null,{default:l(()=>[e("$")]),_:1}),e(),w,I,e(">header1|1 "),y,e("9606 9606 2759 9606 9606 9606 9606 9606 9606 9606 8287 "),B,e(">header1|2 "),V,e("2026807 888268 186802 1598 1883 "),k,e(">header1|3 "),v,e("1883 "),D,e(">header1|1R "),P,e("27342 2759 155619 1133106 38033 2 "),S,e(">header1|2R "),$,e(">header1|3R "),F,e("2951 "),N,n(s,null,{default:l(()=>[e("$")]),_:1}),e(),R,H,e(">header1|1 "),z,e("9606 9606 2759 9606 9606 9606 9606 9606 9606 9606 8287 ")]),_:1}),M,n(d,{class:"mt-5 mb-3",parameters:i})]),_:1})]),_:1}))}});export{ee as default};
