import{I as o}from"./InlineCode-b941903d.js";import{H as r}from"./HeaderBodyCard-f3b47b61.js";import{I as a}from"./Initialism-25eec5f3.js";import{B as u,S as l}from"./Boxed-1ca540a2.js";import{_ as d}from"./ParameterTable.vue_vue_type_script_setup_true_lang-ea4e3ff2.js";import{d as c,o as p,c as _,w as n,l as m,a as s,g as t,b as e}from"./index-fa2d734a.js";import{V as h}from"./VDivider-7dfa9365.js";import"./_plugin-vue_export-helper-c27b6911.js";import"./VCard-4d112d66.js";import"./VRow-781382bd.js";const f=t("h1",{class:"font-weight-light"}," umgap snaptaxon ",-1),b=t("h3",{class:"font-weight-light"}," Snaps taxon IDs to a rank or specified taxa ",-1),g=t("i",null,"standard input",-1),x=t("i",null,"standard output",-1),y=t("b",null,"cat input.fa",-1),v=t("br",null,null,-1),k=t("br",null,null,-1),I=t("br",null,null,-1),T=t("br",null,null,-1),w=t("br",null,null,-1),B=t("br",null,null,-1),V=t("b",null,"umgap snaptaxon 2020-04-taxons.tsv -r order < ~/input.fa",-1),C=t("br",null,null,-1),S=t("br",null,null,-1),A=t("br",null,null,-1),D=t("br",null,null,-1),N=t("br",null,null,-1),P=t("br",null,null,-1),$=t("b",null,"umgap snaptaxon 2020-04-taxons.tsv -t 1239 2 < ~/input.fa",-1),F=t("br",null,null,-1),H=t("br",null,null,-1),j=t("br",null,null,-1),q=t("br",null,null,-1),M=t("br",null,null,-1),Z=c({__name:"MetagenomicsSnapTaxonPage",setup(O){const i=[{name:"-h / --help",description:"Prints help information"},{name:"-V / --version",description:"Prints version information"},{name:"-i / --invalid",description:"Include the invalidated taxa from the taxonomy"},{name:"-r / --rank r",description:"The rank to snap towards [possible values: superkingdom, kingdom, subkingdom, superphylum, phylum, subphylum, superclass, class, subclass, infraclass, superorder, order, suborder, infraorder, parvorder, superfamily, family, subfamily, tribe, subtribe, genus, subgenus, species group, species subgroup, species, subspecies, varietas, forma]"},{name:"-t / --taxons t...",description:"A taxon to snap towards (allow multiple times)"}];return(z,E)=>(p(),_(m,null,{default:n(()=>[f,b,s(h,{class:"my-2"}),t("p",null,[e(" The "),s(o,null,{default:n(()=>[e("umgap snaptaxon")]),_:1}),e(" command takes one or more taxon IDs. For each taxon, it searches amongst its ancestors if any are of the specified rank (e.g., "),s(o,null,{default:n(()=>[e("-r species")]),_:1}),e("), or are one of the listed taxa (e.g., "),s(o,null,{default:n(()=>[e("-t 1598 -t 1883")]),_:1}),e("). If so, the taxon is replaced by the most specific of these matching taxa. Otherwise, it is mapped to the root of the taxonomy. ")]),s(r,{id:"usage",title:"usage","large-title":""},{default:n(()=>[t("p",null,[e(" The input is given on "),g,e(" and may be any sequence of "),s(a,null,{default:n(()=>[e("FASTA")]),_:1}),e(" headers and/or lines containing a single taxon ID. The "),s(a,null,{default:n(()=>[e("FASTA")]),_:1}),e(" headers (if any) are just copied over to "),x,e(". ")]),t("p",null,[e(" The taxonomy to be used is passed as an argument to this command. This is a preprocessed version of the "),s(a,null,{default:n(()=>[e("NCBI")]),_:1}),e(" taxonomy. ")]),s(u,{style:{"max-height":"fit-content"}},{default:n(()=>[s(l,null,{default:n(()=>[e("$")]),_:1}),e(),y,v,e(">header1 "),k,e("888268 "),I,e("186802 "),T,e("1598 "),w,e("1883 "),B,s(l,null,{default:n(()=>[e("$")]),_:1}),e(),V,C,e(">header1 "),S,e("38820 "),A,e("186802 "),D,e("186826 "),N,e("85011 "),P,s(l,null,{default:n(()=>[e("$")]),_:1}),e(),$,F,e(">header1 "),H,e("1 "),j,e("1239 "),q,e("1239 "),M,e("2 ")]),_:1}),s(d,{class:"mt-5 mb-3",parameters:i})]),_:1})]),_:1}))}});export{Z as default};
