import{H as c}from"./HeaderBodyCard-f3b47b61.js";import{R as o}from"./ResourceLink-82ddf6bf.js";import{B as r,S as i}from"./Boxed-1ca540a2.js";import{I as h}from"./InlineCode-b941903d.js";import{I as n}from"./Initialism-25eec5f3.js";import{_ as g}from"./ImageCaptionCard.vue_vue_type_script_setup_true_lang-dcd0da81.js";import{d as p,o as R,c as b,w as a,l as m,g as e,b as t,a as l,t as f,u as E,e as z,f as v,F as y,p as q,q as k}from"./index-fa2d734a.js";import{_ as w}from"./_plugin-vue_export-helper-c27b6911.js";import"./VCard-4d112d66.js";const x="/unipept/assets/casestudy-umgap-comparative-950860c5.svg",s=u=>(q("data-v-34a999e1"),u=u(),k(),u),C=s(()=>e("h1",{class:"font-weight-light"}," Case Study: Running a comparative analysis on 8 samples ",-1)),O={class:"font-weight-light"},B=s(()=>e("br",null,null,-1)),P=s(()=>e("br",null,null,-1)),S=s(()=>e("br",null,null,-1)),M=s(()=>e("br",null,null,-1)),I=s(()=>e("br",null,null,-1)),T=s(()=>e("br",null,null,-1)),F=s(()=>e("br",null,null,-1)),A=s(()=>e("br",null,null,-1)),G=s(()=>e("br",null,null,-1)),V=s(()=>e("br",null,null,-1)),U=s(()=>e("br",null,null,-1)),$=s(()=>e("br",null,null,-1)),N=s(()=>e("br",null,null,-1)),W=s(()=>e("br",null,null,-1)),D=s(()=>e("br",null,null,-1)),j=s(()=>e("br",null,null,-1)),H=s(()=>e("p",null," The next step and most time-consuming step is the analysis. We'll analyse all samples in a single run so we must only load the database in memory once. ",-1)),J=s(()=>e("br",null,null,-1)),K=s(()=>e("br",null,null,-1)),L=s(()=>e("br",null,null,-1)),Z=s(()=>e("br",null,null,-1)),Q=s(()=>e("br",null,null,-1)),X=s(()=>e("br",null,null,-1)),Y=s(()=>e("br",null,null,-1)),tt=s(()=>e("br",null,null,-1)),et=s(()=>e("br",null,null,-1)),st=s(()=>e("br",null,null,-1)),at=s(()=>e("br",null,null,-1)),lt=s(()=>e("br",null,null,-1)),ot=s(()=>e("br",null,null,-1)),nt=s(()=>e("br",null,null,-1)),it=s(()=>e("br",null,null,-1)),ut=s(()=>e("br",null,null,-1)),dt=s(()=>e("br",null,null,-1)),rt=s(()=>e("br",null,null,-1)),ct=s(()=>e("br",null,null,-1)),ft=s(()=>e("br",null,null,-1)),_t=s(()=>e("br",null,null,-1)),Rt=s(()=>e("br",null,null,-1)),ht=s(()=>e("br",null,null,-1)),gt=s(()=>e("br",null,null,-1)),pt=s(()=>e("br",null,null,-1)),bt=s(()=>e("br",null,null,-1)),mt=s(()=>e("br",null,null,-1)),Et=s(()=>e("h2",{class:"font-weight-light mt-5"}," References ",-1)),zt=p({__name:"MetagenomicsComparativePage",setup(u){const _={detender:{full:"De Tender, C., Mesuere, B., Van der Jeugt, F. et al. Peat substrate amended with chitin modulates the N-cycle, siderophore and chitinase responses in the lettuce rhizobiome. Sci Rep 9, 9890 (2019).",short:"De Tender, 2019",link:"doi.org/10.1038/s41598-019-46106-x"}};return(vt,yt)=>(R(),b(m,null,{default:a(()=>[C,e("h3",O,[t(" This case study shows how to run a comparative diversity analysis on 8 metagenomics samples using "),l(n,null,{default:a(()=>[t("UMGAP")]),_:1}),t(". ")]),l(c,{class:"mt-5",title:"Setup","large-title":""},{default:a(()=>[e("p",null,[t(" This case study effectively repeats an analysis of 8 shotgun metagenomics samples and the comparison of the taxonomic diversity therein ("+f(_.detender.short)+"), with a more recent version of the UniprotKB and ",1),l(n,null,{default:a(()=>[t("UMGAP")]),_:1}),t(". ")]),e("p",null,[t(" The raw sequence data of the shotgun metagenomics sequencing is freely available at "),l(o,{to:"https://www.ebi.ac.uk/ena/browser/view/PRJEB15448"},{default:a(()=>[t("ENA")]),_:1}),t(" under the project "),l(n,null,{default:a(()=>[t("ERP017180")]),_:1}),t(". We will start our casestudy by downloading these files manually or via the command line. ")]),l(r,null,{default:a(()=>[l(i,null,{default:a(()=>[t("$")]),_:1}),t(" mkdir casestudy && cd casestudy "),B,l(i,null,{default:a(()=>[t("$")]),_:1}),t(" curl -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/009/ERR1654119/ERR1654119_1.fastq.gz \\ "),P,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/009/ERR1654119/ERR1654119_2.fastq.gz \\ "),S,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/000/ERR1654120/ERR1654120_1.fastq.gz \\ "),M,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/000/ERR1654120/ERR1654120_2.fastq.gz \\ "),I,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/001/ERR1654121/ERR1654121_1.fastq.gz \\ "),T,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/001/ERR1654121/ERR1654121_2.fastq.gz \\ "),F,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/002/ERR1654122/ERR1654122_1.fastq.gz \\ "),A,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/002/ERR1654122/ERR1654122_2.fastq.gz \\ "),G,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/003/ERR1654123/ERR1654123_1.fastq.gz \\ "),V,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/003/ERR1654123/ERR1654123_2.fastq.gz \\ "),U,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/004/ERR1654124/ERR1654124_1.fastq.gz \\ "),$,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/004/ERR1654124/ERR1654124_2.fastq.gz \\ "),N,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/005/ERR1654125/ERR1654125_1.fastq.gz \\ "),W,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/005/ERR1654125/ERR1654125_2.fastq.gz \\ "),D,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/006/ERR1654126/ERR1654126_1.fastq.gz \\ "),j,t(" -O ftp.sra.ebi.ac.uk/vol1/fastq/ERR165/006/ERR1654126/ERR1654126_2.fastq.gz ")]),_:1}),e("p",null,[t(" While these files are downloading (combined size is 42GB), you can install and configure "),l(n,null,{default:a(()=>[t("UMGAP")]),_:1}),t(". For this case study, we'll need the "),l(n,null,{default:a(()=>[t("NCBI")]),_:1}),t(' taxonomy and the 9-mer index file (152GB), so answer "yes" when the setup script asks for those. ')])]),_:1}),l(c,{class:"mt-5",title:"Analysis","large-title":""},{default:a(()=>[H,l(r,null,{default:a(()=>[l(i,null,{default:a(()=>[t("$")]),_:1}),t(" umgap-analyse.sh \\ "),J,t(" -1 ERR1654119_1.fastq.gz -2 ERR1654119_2.fastq.gz -t high-sensitivity -z -o ERR1654119.tid.fa.gz \\ "),K,t(" -1 ERR1654120_1.fastq.gz -2 ERR1654120_2.fastq.gz -t high-sensitivity -z -o ERR1654120.tid.fa.gz \\ "),L,t(" -1 ERR1654121_1.fastq.gz -2 ERR1654121_2.fastq.gz -t high-sensitivity -z -o ERR1654121.tid.fa.gz \\ "),Z,t(" -1 ERR1654122_1.fastq.gz -2 ERR1654122_2.fastq.gz -t high-sensitivity -z -o ERR1654122.tid.fa.gz \\ "),Q,t(" -1 ERR1654123_1.fastq.gz -2 ERR1654123_2.fastq.gz -t high-sensitivity -z -o ERR1654123.tid.fa.gz \\ "),X,t(" -1 ERR1654124_1.fastq.gz -2 ERR1654124_2.fastq.gz -t high-sensitivity -z -o ERR1654124.tid.fa.gz \\ "),Y,t(" -1 ERR1654125_1.fastq.gz -2 ERR1654125_2.fastq.gz -t high-sensitivity -z -o ERR1654125.tid.fa.gz \\ "),tt,t(" -1 ERR1654126_1.fastq.gz -2 ERR1654126_2.fastq.gz -t high-sensitivity -z -o ERR1654126.tid.fa.gz ")]),_:1}),e("p",null,[t(" We can analyse the output files ("),l(h,null,{default:a(()=>[t("*.tid.fa.gz")]),_:1}),t(") individually or comparatively. For the first, "),l(n,null,{default:a(()=>[t("UMGAP")]),_:1}),t(" includes a command to create interactive visualisations. For each output file, we get a link. ")]),l(r,null,{default:a(()=>[l(i,null,{default:a(()=>[t("$")]),_:1}),t(" umgap-visualize.sh -u *.tid.fa.gz "),et,t("ERR1654119.tid.fa.gz: "),l(o,{to:"https://bl.ocks.org/5055456c85e9e103da221fbba6e6c269"},{default:a(()=>[t("https://bl.ocks.org/5055456c85e9e103da221fbba6e6c269")]),_:1}),st,t("ERR1654120.tid.fa.gz: "),l(o,{to:"https://bl.ocks.org/31e0fc6e2d8705e5ef64e856318dc17c"},{default:a(()=>[t("https://bl.ocks.org/31e0fc6e2d8705e5ef64e856318dc17c")]),_:1}),at,t("ERR1654121.tid.fa.gz: "),l(o,{to:"https://bl.ocks.org/019b89c96446a2949423d34f4281ee1c"},{default:a(()=>[t("https://bl.ocks.org/019b89c96446a2949423d34f4281ee1c")]),_:1}),lt,t("ERR1654122.tid.fa.gz: "),l(o,{to:"https://bl.ocks.org/3542a2a24fcf5ed3a791e937d2d6e618"},{default:a(()=>[t("https://bl.ocks.org/3542a2a24fcf5ed3a791e937d2d6e618")]),_:1}),ot,t("ERR1654123.tid.fa.gz: "),l(o,{to:"https://bl.ocks.org/ccc88f795dbd797b08cfe7c8d1187a13"},{default:a(()=>[t("https://bl.ocks.org/ccc88f795dbd797b08cfe7c8d1187a13")]),_:1}),nt,t("ERR1654124.tid.fa.gz: "),l(o,{to:"https://bl.ocks.org/794f9907fbb2488d94f94ccb7d06d835"},{default:a(()=>[t("https://bl.ocks.org/794f9907fbb2488d94f94ccb7d06d835")]),_:1}),it,t("ERR1654125.tid.fa.gz: "),l(o,{to:"https://bl.ocks.org/7568952f1338768a6b62d32928c0ee74"},{default:a(()=>[t("https://bl.ocks.org/7568952f1338768a6b62d32928c0ee74")]),_:1}),ut,t("ERR1654126.tid.fa.gz: "),l(o,{to:"https://bl.ocks.org/f25941840ce1e706623750d561d7c47d"},{default:a(()=>[t("https://bl.ocks.org/f25941840ce1e706623750d561d7c47d")]),_:1})]),_:1})]),_:1}),l(c,{class:"mt-5",title:"Visualisation","large-title":""},{default:a(()=>[e("p",null,[t(" The 8 analysed samples are, however, related. We would like to have comparative results. "),l(n,null,{default:a(()=>[t("UMGAP")]),_:1}),t(" does not support such visualisations directly, but all results can be exported to a comparative "),l(n,null,{default:a(()=>[t("CSV")]),_:1}),t(" frequency table at a rank of choice, e.g. genus or phylum, as below. ")]),l(r,null,{default:a(()=>[l(i,null,{default:a(()=>[t("$")]),_:1}),t(" umgap-visualize.sh -r phylum -t *.tid.fa.gz > frequencies.csv "),dt,l(i,null,{default:a(()=>[t("$")]),_:1}),t(" head -10 frequencies.csv "),rt,t("taxon id,taxon name,ERR1654119.taxa.fasta.gz,ERR1654120.taxa.fasta.gz,ERR1654121.taxa.fasta.gz,ERR1654122.taxa.fasta.gz,ERR1654123.taxa.fasta.gz,ERR1654124.taxa.fasta.gz,ERR1654125.taxa.fasta.gz,ERR1654126.taxa.fasta.gz "),ct,t("1239,Firmicutes,83406,150700,80856,174901,80888,78400,95509,132121 "),ft,t("2497438,Perkinsozoa,161,284,122,285,148,135,171,267 "),_t,t("1752735,Candidatus Wolfebacteria,113,209,108,231,124,82,109,162 "),Rt,t("1817902,Candidatus Brennerbacteria,4,5,4,5,2,4,6,1 "),ht,t("221216,Candidatus Parcubacteria,734,1485,823,1646,914,935,1304,1788 "),gt,t("1817898,Candidatus Blackburnbacteria,28,66,26,57,18,41,49,68 "),pt,t("3041,Chlorophyta,6439,11167,5921,13904,6303,6138,6897,10420 "),bt,t("1913638,Zoopagomycota,1261,1730,937,2292,9727,6500,6728,7377 "),mt,t("2732406,Kitrinoviricota,38,76,59,102,43,37,60,86 ")]),_:1}),e("p",null,[t(" This "),l(n,null,{default:a(()=>[t("CSV")]),_:1}),t(" file can be processed by your favorite spreadsheet software or statistical analysis frameworks such as R (packages of interest are "),l(o,{to:"https://joey711.github.io/phyloseq/"},{default:a(()=>[t("phyloseq")]),_:1}),t(" and "),l(o,{to:"https://cran.rstudio.com/web/packages/vegan/index.html"},{default:a(()=>[t("vegan")]),_:1}),t(", and "),l(o,{to:"https://ggplot2.tidyverse.org/"},{default:a(()=>[t("ggplot2")]),_:1}),t(" for visualizations). For this study, it's interesting to have a look at the fraction each of the 20 most detected phyla represents in its sample. For phyla uninfluenced by the addition of chitin to the sample, we expect to see a uniform spread over all 8 samples. The Mucoromycota and Ciliophora phyla, however, are overrepresented in the 4 chitin-rich samples. ")]),l(g,{class:"mt-5",image:E(x)},null,8,["image"])]),_:1}),Et,e("ul",null,[(R(),z(y,null,v(_,d=>e("li",{key:d.short},[t(f(d.full)+" ",1),l(o,{to:"https://"+d.link},{default:a(()=>[t(f(d.link),1)]),_:2},1032,["to"])])),64))])]),_:1}))}});const Mt=w(zt,[["__scopeId","data-v-34a999e1"]]);export{Mt as default};
