import{I as a}from"./InlineCode-67d03729.js";import{H as u}from"./HeaderBodyCard-3295d4c5.js";import{I as s}from"./Initialism-440d96b0.js";import{B as i,S as l}from"./Boxed-9f9dd66a.js";import{_ as d}from"./ParameterTable.vue_vue_type_script_setup_true_lang-eb31a2e0.js";import{d as c,o as _,c as h,w as o,l as m,a as e,g as n,b as t}from"./index-eea28a33.js";import{V as p}from"./VDivider-834b9f8e.js";import"./_plugin-vue_export-helper-c27b6911.js";import"./VCard-9d59c194.js";import"./VRow-94e3f3e6.js";const f=n("h1",{class:"font-weight-light"}," umgap taxa2freq ",-1),x=n("h3",{class:"font-weight-light"}," Counts ranked taxon occurrences in a stream of taxon IDs or arguments ",-1),b=n("i",null,"standard input",-1),g=n("b",null,"cat input.txt",-1),k=n("br",null,null,-1),y=n("br",null,null,-1),v=n("br",null,null,-1),q=n("br",null,null,-1),C=n("br",null,null,-1),T=n("br",null,null,-1),w=n("br",null,null,-1),I=n("br",null,null,-1),V=n("br",null,null,-1),B=n("br",null,null,-1),$=n("br",null,null,-1),D=n("br",null,null,-1),H=n("b",null,"umgap taxa2freq taxons.tsv < input.txt",-1),N=n("br",null,null,-1),P=n("br",null,null,-1),S=n("br",null,null,-1),E=n("br",null,null,-1),W=n("b",null,"umgap taxa2freq taxons.tsv input.txt input.txt",-1),F=n("br",null,null,-1),M=n("br",null,null,-1),j=n("br",null,null,-1),z=n("b",null,"umgap taxa2freq -r phylum taxons.tsv < input.txt",-1),A=n("br",null,null,-1),G=n("br",null,null,-1),ot=c({__name:"MetagenomicsTaxa2FreqPage",setup(J){const r=[{name:"-h / --help",description:"Prints help information"},{name:"-V / --version",description:"Prints version information"},{name:"-f / --frequency f",description:"The minimum frequency to be reported [default: 1]"},{name:"-r / --rank r",description:"The rank to show [default: species]"}];return(K,L)=>(_(),h(m,null,{default:o(()=>[f,x,e(p,{class:"my-2"}),n("p",null,[t(" The "),e(a,null,{default:o(()=>[t("umgap taxa2freq")]),_:1}),t(" command creates a frequency table of a list of taxa on a given target rank (species by default). When invoked with file arguments, it adds a column for each file. ")]),e(u,{id:"usage",title:"usage","large-title":""},{default:o(()=>[n("p",null,[t(" The input is given on "),b,t(", or in multiple file arguments, a single taxon ID on each line. Each taxon that is more specific than the target rank is counted towards its ancestor on the target rank. Each taxon less specific than the target rank is counted towards root. The command outputs a "),e(s,null,{default:o(()=>[t("CSV")]),_:1}),t(" table of counts, taxon IDs and their names. ")]),n("p",null,[t(" The taxonomy to be used is passed as an argument to this command. This is a preprocessed version of the "),e(s,null,{default:o(()=>[t("NCBI")]),_:1}),t(" taxonomy. ")]),e(i,{style:{"max-height":"fit-content"}},{default:o(()=>[e(l,null,{default:o(()=>[t("$")]),_:1}),t(),g,k,t("9606 "),y,t("9606 "),v,t("2759 "),q,t("9606 "),C,t("9606 "),T,t("9606 "),w,t("9606 "),I,t("9606 "),V,t("9606 "),B,t("9606 "),$,t("8287 "),D,e(l,null,{default:o(()=>[t("$")]),_:1}),t(),H,N,t("taxon id,taxon name,stdin "),P,t("1,root,2 "),S,t("9606,Homo sapiens,9 "),E,e(l,null,{default:o(()=>[t("$")]),_:1}),t(),W,F,t("taxon id,taxon name,input.txt,input.txt "),M,t("1,root,2,2 "),j,t("9606,Homo sapiens,9,9 ")]),_:1}),n("p",null,[t(" With the "),e(a,null,{default:o(()=>[t("-r")]),_:1}),t(" option, the default species rank can be set to any named rank. ")]),e(i,null,{default:o(()=>[e(l,null,{default:o(()=>[t("$")]),_:1}),t(),z,A,t("taxon id,taxon name,stdin "),G,t("7711,Chordata,10 ")]),_:1}),e(d,{class:"mt-5 mb-3",parameters:r})]),_:1})]),_:1}))}});export{ot as default};
