import{R as l}from"./ResourceLink-82ddf6bf.js";import{V as w,b as c,a as _}from"./VCard-4d112d66.js";import{V as P}from"./VDivider-7dfa9365.js";import{a as U,V as d}from"./VRow-781382bd.js";import{d as m,o as h,c as f,w as t,a,g as s,t as p,r as x,b as e,ar as u,m as I,V,l as T,u as n}from"./index-fa2d734a.js";import{V as B}from"./VHover-fb20754c.js";import{I as o}from"./Initialism-25eec5f3.js";import"./_plugin-vue_export-helper-c27b6911.js";import"./delay-fab8d36b.js";const j={style:{"font-size":"1.25rem"}},M=s("h2",{class:"mb-3"}," Abstract ",-1),D=s("h2",{class:"mt-5 mb-3"}," Citation ",-1),z=s("br",null,null,-1),C=s("br",null,null,-1),A=s("br",null,null,-1),r=m({__name:"PublicationCard",props:{title:{},authors:{},journal:{},year:{},extra:{default:""},doi:{},googleScholar:{},image:{}},setup(g){return(i,b)=>(h(),f(c,null,{default:t(()=>[a(w,{class:"bg-blue text-white",style:{"letter-spacing":"0.0125rem"}},{default:t(()=>[s("div",j,p(i.title),1),s("div",null,p(i.authors.length==1?i.authors[0]:i.authors.slice(0,-1).join(", ")+", and "+i.authors[i.authors.length-1]),1)]),_:1}),a(P),a(_,null,{default:t(()=>[a(U,null,{default:t(()=>[a(d,{cols:"12",md:"8",xl:"9"},{default:t(()=>[M,x(i.$slots,"abstract"),D,e(" "+p(i.authors.length==1?i.authors[0]:i.authors.slice(0,-1).join(", ")+", and "+i.authors[i.authors.length-1])+" ",1),z,s("b",null,p(i.title),1),C,e(" "+p(i.journal+", "+i.year+(i.extra?", "+i.extra:i.extra))+" ",1),A,s("div",null,[a(u,{size:"x-small",class:"mr-1"},{default:t(()=>[e(" mdi-link-variant ")]),_:1}),a(l,{to:"https://"+i.doi},{default:t(()=>[e(p(i.doi),1)]),_:1},8,["to"]),a(u,{class:"ml-1",size:"x-small"},{default:t(()=>[e(" mdi-link-variant ")]),_:1}),a(l,{to:i.googleScholar},{default:t(()=>[e(" Google Scholar ")]),_:1},8,["to"])])]),_:3}),a(d,{cols:"12",md:"4",xl:"3"},{default:t(()=>[a(l,{to:"https://"+i.doi},{default:t(()=>[a(B,null,{default:t(({isHovering:y,props:v})=>[a(c,I({elevation:y?6:2},v),{default:t(()=>[a(V,{src:i.image,alt:i.image},null,8,["src","alt"])]),_:2},1040,["elevation"])]),_:1})]),_:1},8,["to"])]),_:1})]),_:3})]),_:3})]),_:3}))}}),k="/unipept/assets/publications-2021-98b6e50a.jpg",S="/unipept/assets/publications-2020-51393a04.jpg",L="/unipept/assets/publications-2019-b8cebde3.jpg",q="/unipept/assets/publications-2018-547d5217.jpg",J="/unipept/assets/publications-2016b-3b7ada0b.jpg",F="/unipept/assets/publications-2016a-7ec8ccf6.jpg",N="/unipept/assets/publications-2015-ac98effd.jpg",R="/unipept/assets/publications-2012-0c29cb31.jpg",G=s("h1",{class:"font-weight-light"}," Publications ",-1),$=s("h3",{class:"font-weight-light"}," This page lists all Unipept peer-reviewed journal articles in reverse-chronological order. ",-1),ee=m({__name:"PublicationsPage",setup(g){return(i,b)=>(h(),f(T,null,{default:t(()=>[G,$,a(r,{class:"mt-5",title:"Unipept Desktop: a faster, more powerful metaproteomics results analysis tool",authors:["Pieter Verschaffelt","Tim Van Den Bossche","Lennart Martens","Peter Dawyndt","Bart Mesuere"],journal:"Journal of Proteome Research",year:"2021",doi:"doi.org/10.1021/acs.jproteome.0c00855","google-scholar":"https://scholar.google.com/scholar?cluster=14287692748263471832&hl=nl&as_sdt=0,5",image:n(k)},{abstract:t(()=>[e(" Metaproteomics has become an important research tool to study microbial systems, which has resulted in increased metaproteomics data generation. However, efficient tools for processing the acquired data have lagged behind. One widely used tool for metaproteomics data interpretation is Unipept, a web-based tool that provides, among others, interactive and insightful visualizations. Due to its web-based implementation, however, the Unipept web application is limited in the amount of data that can be analyzed. In this manuscript we therefore present Unipept Desktop, a desktop application version of Unipept that is designed to drastically increase the throughput and capacity of metaproteomics data analysis. Moreover, it provides a novel comparative analysis pipeline and improves the organization of experimental data into projects, thus addressing the growing need for more efficient and versatile analysis tools for metaproteomics data. ")]),_:1},8,["image"]),a(r,{class:"mt-5",title:"Unipept CLI 2.0: adding support for visualizations and functional annotations",authors:["Pieter Verschaffelt","Philippe Van Thienen","Tim Van Den Bossche","Felix Van der Jeugt","Caroline De Tender","Lennart Martens","Peter Dawyndt","Bart Mesuere"],journal:"Bioinformatics",year:"2020",doi:"doi.org/10.1093/bioinformatics/btaa553","google-scholar":"https://scholar.google.com/scholar?cluster=12996336077971755665&hl=nl&as_sdt=0,5",image:n(S)},{abstract:t(()=>[e(" Unipept is an ecosystem of tools developed for fast metaproteomics data-analysis consisting of a web application, a set of web services (application programming interface, "),a(o,null,{default:t(()=>[e("API")]),_:1}),e(") and a command-line interface ("),a(o,null,{default:t(()=>[e("CLI")]),_:1}),e("). After the successful introduction of version 4 of the Unipept web application, we here introduce version 2.0 of the "),a(o,null,{default:t(()=>[e("API")]),_:1}),e(" and "),a(o,null,{default:t(()=>[e("CLI")]),_:1}),e(". Next to the existing taxonomic analysis, version 2.0 of the "),a(o,null,{default:t(()=>[e("API")]),_:1}),e(" and "),a(o,null,{default:t(()=>[e("CLI")]),_:1}),e(" provides access to Unipept’s powerful functional analysis for metaproteomics samples. The functional analysis pipeline supports retrieval of Enzyme Commission numbers, Gene Ontology terms and InterPro entries for the individual peptides in a metaproteomics sample. This paves the way for other applications and developers to integrate these new information sources into their data processing pipelines, which greatly increases insight into the functions performed by the organisms in a specific environment. Both the "),a(o,null,{default:t(()=>[e("API")]),_:1}),e(" and "),a(o,null,{default:t(()=>[e("CLI")]),_:1}),e(" have also been expanded with the ability to render interactive visualizations from a list of taxon ids. These visualizations are automatically made available on a dedicated website and can easily be shared by users. ")]),_:1},8,["image"]),a(r,{class:"mt-5",title:"Unipept 4.0: functional analysis of metaproteome data",authors:["Robbert Gurdeep Singh","Alessandro Tanca","Antonio Palomba","Felix Van der Jeugt","Pieter Verschaffelt","Sergio Uzzau","Lennart Martens","Peter Dawyndt","Bart Mesuere"],journal:"Journal of Proteome Research",year:"2019",doi:"doi.org/10.1021/acs.jproteome.8b00716","google-scholar":"https://scholar.google.be/scholar?cluster=10704911090193210592&hl=en&as_sdt=0,5",image:n(L)},{abstract:t(()=>[e(" Unipept is a web application for metaproteome data analysis, with an initial focus on tryptic-peptide-based biodiversity analysis of "),a(o,null,{default:t(()=>[e("MS/MS")]),_:1}),e(" samples. Because the true potential of metaproteomics lies in gaining insight into the expressed functions of complex environmental samples, the 4.0 release of Unipept introduces complementary functional analysis based on "),a(o,null,{default:t(()=>[e("GO")]),_:1}),e(" terms and "),a(o,null,{default:t(()=>[e("EC")]),_:1}),e(" numbers. Integration of this new functional analysis with the existing biodiversity analysis is an important asset of the extended pipeline. As a proof of concept, a human faecal metaproteome data set from 15 healthy subjects was reanalyzed with Unipept 4.0, yielding fast, detailed, and straightforward characterization of taxon-specific catalytic functions that is shown to be consistent with previous results from a "),a(o,null,{default:t(()=>[e("BLAST")]),_:1}),e("-based functional analysis of the same data. ")]),_:1},8,["image"]),a(r,{class:"mt-5",title:"High-throughput metaproteomics data analysis with Unipept: a tutorial",authors:["Bart Mesuere","Felix Van der Jeugt","Toon Willems","Tom Naessens","Bart Devreese","Lennart Martens","Peter Dawyndt"],journal:"Journal of Proteomics",year:"2018",extra:"171, pp 11-22",doi:"doi.org/10.1016/j.jprot.2017.05.022","google-scholar":"https://scholar.google.be/scholar?cluster=6566003157091239958&hl=en&as_sdt=0,5",image:n(q)},{abstract:t(()=>[e(" In recent years, shotgun metaproteomics has established itself as an important tool to study the composition of complex ecosystems and microbial communities. Two key steps in metaproteomics data analysis are the inference of proteins from the identified peptides, and the determination of the taxonomic origin and function of these proteins. This tutorial therefore introduces the Unipept command line interface ("),a(l,{to:"/clidocs",router:""},{default:t(()=>[e("http://unipept.ugent.be/clidocs")]),_:1}),e(" ) as a platform-independent tool for such metaproteomics data analyses. First, a detailed overview is given of the available Unipept commands and their functions. Next, the power of the Unipept command line interface is illustrated using two case studies that analyze a single tryptic peptide, and a set of peptides retrieved from a shotgun metaproteomics experiment, respectively. Finally, the analysis results obtained using these command line tools are compared with the interactive taxonomic analysis that is available on the Unipept website. ")]),_:1},8,["image"]),a(r,{class:"mt-5",title:"The unique peptidome: Taxon-specific tryptic peptides as biomarkers for targeted metaproteomics",authors:["Bart Mesuere","Felix Van der Jeugt","Bart Devreese","Peter Vandamme","Peter Dawyndt"],journal:"Proteomics",year:"2016",extra:"16 (17), pp 2313–2318",doi:"doi.org/10.1002/pmic.201600023","google-scholar":"https://scholar.google.com/scholar?cluster=10033728957067620724&hl=en&as_sdt=0,5",image:n(J)},{abstract:t(()=>[e(" The Unique Peptide Finder is an interactive web application to quickly hunt for tryptic peptides that are unique to a particular species, genus, or any other taxon. Biodiversity within the target taxon is represented by a set of proteomes selected from a monthly updated list of complete and nonredundant UniProt proteomes, supplemented with proprietary proteomes loaded into persistent local browser storage. The software computes and visualizes pan and core peptidomes as unions and intersections of tryptic peptides occurring in the selected proteomes. In addition, it also computes and displays unique peptidomes as the set of all tryptic peptides that occur in all selected proteomes but not in any UniProt record not assigned to the target taxon. As a result, the unique peptides can serve as robust biomarkers for the target taxon, for example, in targeted metaproteomics studies. Computations are extremely fast since they are underpinned by the Unipept database, the lowest common ancestor algorithm implemented in Unipept and modern web technologies that facilitate in-browser data storage and parallel processing. ")]),_:1},8,["image"]),a(r,{class:"mt-5",title:"Unipept Web Services for Metaproteomics Analysis",authors:["Bart Mesuere","Toon Willems","Felix Van der Jeugt","Bart Devreese","Peter Vandamme","Peter Dawyndt"],journal:"Bioinformatics",year:"2016",extra:"32 (11), pp 1746-1748",doi:"doi.org/10.1093/bioinformatics/btw039","google-scholar":"https://scholar.google.be/scholar?cluster=6023517943371055318&hl=en&as_sdt=0,5",image:n(F)},{abstract:t(()=>[e(" Unipept is an open source web application that is designed for metaproteomics analysis with a focus on interactive datavisualization. It is underpinned by a fast index built from UniProtKB and the "),a(o,null,{default:t(()=>[e("NCBI")]),_:1}),e(" taxonomy that enables quick retrieval of all UniProt entries in which a given tryptic peptide occurs. Unipept version 2.4 introduced web services that provide programmatic access to the metaproteomics analysis features. This enables integration of Unipept functionality in custom applications and data processing pipelines. Availability and Implementation The web services are freely available at "),a(l,{to:"http://api.unipept.ugent.be"},{default:t(()=>[e("http://api.unipept.ugent.be")]),_:1}),e("> and are open sourced under the "),a(o,null,{default:t(()=>[e("MIT")]),_:1}),e(" license. ")]),_:1},8,["image"]),a(r,{class:"mt-5",title:"The Unipept Metaproteomics Analysis Pipeline",authors:["Bart Mesuere","Griet Debyser","Maarten Aerts","Bart Devreese","Peter Vandamme","Peter Dawyndt"],journal:"Proteomics",year:"2015",extra:"15 (8), pp 1437–1442",doi:"doi.org/10.1002/pmic.201400361","google-scholar":"https://scholar.google.be/scholar?cluster=4081421507601531351&hl=en&as_sdt=0,5",image:n(N)},{abstract:t(()=>[e(" Unipept ("),a(l,{to:"/",router:""},{default:t(()=>[e("http://unipept.ugent.be")]),_:1}),e(") is a web application that offers a user-friendly way to explore the biodiversity of complex metaproteome samples by providing interactive visualizations. In this article, the updates and changes to Unipept since its initial release are presented. This includes the addition of interactive sunburst and treeview visualizations to the multipeptide analysis, the foundations of an application programming interface ("),a(o,null,{default:t(()=>[e("API")]),_:1}),e(") and a command line interface, updated data sources, and the open-sourcing of the entire application under the "),a(o,null,{default:t(()=>[e("MIT")]),_:1}),e(" license. ")]),_:1},8,["image"]),a(r,{class:"mt-5",title:"Unipept: Tryptic Peptide-Based Biodiversity Analysis of Metaproteome Samples",authors:["Bart Mesuere","Bart Devreese","Griet Debyser","Maarten Aerts","Peter Vandamme","Peter Dawyndt"],journal:"J. Proteome Res.",year:"2012",extra:"11 (12), pp 5773–5780",doi:"doi.org/10.1021/pr300576s","google-scholar":"https://scholar.google.be/scholar?cluster=13083723326591523916&hl=en&as_sdt=2005&sciodt=0,5",image:n(R)},{abstract:t(()=>[e(" The Unipept web application ("),a(l,{to:"/",router:""},{default:t(()=>[e("http://unipept.ugent.be")]),_:1}),e(") supports biodiversity analysis of large and complex metaproteome samples using tryptic peptide information obtained from shotgun "),a(o,null,{default:t(()=>[e("MS/MS")]),_:1}),e(" experiments. Its underlying index structure is designed to quickly retrieve all occurrences of a tryptic peptide in UniProt entries. Taxon-specificity of the tryptic peptide is successively derived from these occurrences using a novel lowest common ancestor approach that is robust against taxonomic misarrangements, misidentifications, and inaccuracies. Not taking into account this identification noise would otherwise result in drastic loss of information. Dynamic treemaps visualize the biodiversity of metaproteome samples, which eases the exploration of samples with highly complex compositions. The potential of Unipept to gain novel insights into the biodiversity of a sample is evaluated by reanalyzing publicly available metaproteome data sets taken from the bacterial phyllosphere and the human gut. ")]),_:1},8,["image"])]),_:1}))}});export{ee as default};
